diff --git a/test/e2e/util/util.go b/test/e2e/util/util.go
index aa500cad788..20a8f8496ee 100644
--- a/test/e2e/util/util.go
+++ b/test/e2e/util/util.go
@@ -136,6 +136,11 @@ var (
 		// Allow 1 restart for token-minter sidecar race condition: https://issues.redhat.com/browse/GCP-441
 		// TODO(GCP-447): Remove this toleration once token-minter is injected as a native sidecar init container.
 		"gcp-cloud-controller-manager": 1,
+		// Allow 5 restarts for dns-operator due to new RBAC rollout out by CVO trailing dns-operator rollout
+		// https://redhat.atlassian.net/browse/OCPBUGS-78539
+		// https://redhat.atlassian.net/browse/NE-2500
+		// Can be removed after https://github.com/openshift/cluster-dns-operator/pull/458 is accepted into payload
+		"dns-operator": 5,
 	}
 )
 
