diff --git a/Dockerfile.openshift b/Dockerfile.openshift
index 15a45103..70e6b9a1 100644
--- a/Dockerfile.openshift
+++ b/Dockerfile.openshift
@@ -4,7 +4,7 @@ COPY . .
 RUN make build
 
 FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
-COPY --from=builder /go/bin/ibm-vpc-block-csi-driver /bin/ibm-vpc-block-csi-driver
+COPY --from=builder /go/src/github.com/openshift/ibm-vpc-block-csi-driver/bin/ibm-vpc-block-csi-driver /bin/ibm-vpc-block-csi-driver
 RUN yum install -y util-linux nfs-utils e2fsprogs xfsprogs ca-certificates udev && yum clean all && rm -rf /var/cache/yum
 
 LABEL description="IBM VPC Block CSI Driver"
diff --git a/Makefile b/Makefile
index 15982425..0e68ff26 100644
--- a/Makefile
+++ b/Makefile
@@ -77,7 +77,8 @@ lint: deps
 
 .PHONY: build
 build:
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -mod=vendor -a -ldflags '-X main.vendorVersion='"${DRIVER_NAME}-${GIT_COMMIT_SHA}"' -extldflags "-static"' -o ${GOPATH}/bin/${EXE_DRIVER_NAME} ./cmd/
+# OpenShift: <carry>: do not write the binary to ${GOPATH}, it's not stable in the builder.
+	CGO_ENABLED=0 GOOS=$(shell go env GOOS) GOARCH=$(shell go env GOARCH) go build -mod=vendor -a -ldflags '-X main.vendorVersion='"${DRIVER_NAME}-${GIT_COMMIT_SHA}"' -extldflags "-static"' -o bin/${EXE_DRIVER_NAME} ./cmd/
 
 .PHONY: buildmac
 buildmac:
