diff --git a/test/e2e b/test/e2e
index 9049ac5..25e8740 100755
--- a/test/e2e
+++ b/test/e2e
@@ -141,7 +141,8 @@ test_kernel_version() {
         --rm \
         --image=${rhcos_image} \
         --restart=Never \
-        --command -- rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}" kernel-core)
+        --command -- rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}\n" kernel-core \
+        | head -1)
     echo "INFO: RHCOS kernel: ${rhcos_kernel}"
 
     dtk_release_file=$(get_driver_toolkit_release_file)
