diff --git a/images/router/haproxy/conf/haproxy-config.template b/images/router/haproxy/conf/haproxy-config.template
index b101b5ca0..4d418fc37 100644
--- a/images/router/haproxy/conf/haproxy-config.template
+++ b/images/router/haproxy/conf/haproxy-config.template
@@ -131,6 +131,12 @@ global
     {{- end }}
   {{- end }}
 
+  # By default when a ROUTER_CURVES is not defined HAProxy
+  # will use its built-in default supported groups for TLS key exchange.
+  {{- with (env "ROUTER_CURVES") }}
+  ssl-default-bind-curves {{ . }}
+  {{- end }}
+
 defaults
   {{- with $value := env "ROUTER_MAX_CONNECTIONS" "50000" }}
     {{- if isInteger $value }}
