diff --git a/Dockerfile b/Dockerfile
index b4bf1d6fb..1ff638d28 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,11 @@
 FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
 WORKDIR /go/src/github.com/openshift/route-controller-manager
-COPY . .
+COPY Makefile ./
+COPY go.mod go.sum ./
+COPY .git/ .git/
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+COPY vendor/ vendor/
 RUN make build --warn-undefined-variables
 
 FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
