diff --git a/pkg/controller/incompatible_operator_controller.go b/pkg/controller/incompatible_operator_controller.go
index 617031b5f..106d7d9b0 100644
--- a/pkg/controller/incompatible_operator_controller.go
+++ b/pkg/controller/incompatible_operator_controller.go
@@ -301,7 +301,7 @@ func (c *incompatibleOperatorController) checkIncompatibility(logger logr.Logger
 			if err != nil {
 				err = fmt.Errorf("error converting to semver for version %s: %v", string(p.Value), err)
 				logger.Info(err.Error())
-				errs = append(errs, fmt.Errorf("more than one %s found in bundle", maxOpenShiftVersionProperty))
+				errs = append(errs, err)
 				continue
 			}
 
