diff --git a/pkg/test/extensions/binary.go b/pkg/test/extensions/binary.go
index 1fdf4a13b607..83f5e5d4e5d3 100644
--- a/pkg/test/extensions/binary.go
+++ b/pkg/test/extensions/binary.go
@@ -337,6 +337,10 @@ var extensionBinaries = []TestBinary{
 		imageTag:   "service-ca-operator",
 		binaryPath: "/usr/bin/service-ca-operator-tests-ext.gz",
 	},
+	{
+		imageTag:   "oc-mirror",
+		binaryPath: "/usr/bin/oc-mirror-tests-ext.gz",
+	},
 	{
 		imageTag:   "vsphere-csi-driver-operator",
 		binaryPath: "/usr/bin/vmware-vsphere-csi-driver-operator-tests-ext.gz",
