diff --git a/tools/codegen/pkg/schemacheck/generator.go b/tools/codegen/pkg/schemacheck/generator.go
index bb6d0ca1189..030885ccc4c 100644
--- a/tools/codegen/pkg/schemacheck/generator.go
+++ b/tools/codegen/pkg/schemacheck/generator.go
@@ -204,7 +204,8 @@ type schemaCheckGenerationContext struct {
 // It finds all CRD manifests, loads the data and the original version of the manifest for comparison.
 func loadSchemaCheckGenerationContextsForVersion(version generation.APIVersionContext, gitBaseSHA string) ([]schemaCheckGenerationContext, error) {
 	repo, err := git.PlainOpenWithOptions(version.Path, &git.PlainOpenOptions{
-		DetectDotGit: true,
+		DetectDotGit:          true,
+		EnableDotGitCommonDir: true,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("could not open git repository at %s: %w", version.Path, err)
diff --git a/tools/codegen/pkg/utils/git_diff.go b/tools/codegen/pkg/utils/git_diff.go
index d0d14c72a09..341cec59076 100644
--- a/tools/codegen/pkg/utils/git_diff.go
+++ b/tools/codegen/pkg/utils/git_diff.go
@@ -15,7 +15,8 @@ import (
 // committed and current file contents is returned.
 func GitDiff(packagePath, fileName string) (string, error) {
 	repo, err := git.PlainOpenWithOptions(packagePath, &git.PlainOpenOptions{
-		DetectDotGit: true,
+		DetectDotGit:          true,
+		EnableDotGitCommonDir: true,
 	})
 	if err != nil {
 		return "", fmt.Errorf("error opening parent git repository: %w", err)
