diff --git a/cmd/oc-mirror/main.go b/cmd/oc-mirror/main.go
index 078016b1e6..819b232672 100644
--- a/cmd/oc-mirror/main.go
+++ b/cmd/oc-mirror/main.go
@@ -12,7 +12,6 @@ import (
 	"runtime/pprof"
 	"slices"
 	"strings"
-	"syscall"
 
 	"k8s.io/klog"
 
@@ -83,15 +82,11 @@ func runOcMirrorV1(args []string) error {
 	v1File.Close()
 
 	cmd := exec.Cmd{
-		Path: path,
-		SysProcAttr: &syscall.SysProcAttr{
-			// Kill children if parent is dead
-			Pdeathsig: syscall.SIGKILL,
-			Setpgid:   true,
-		},
-		Args:   args,
-		Stdout: os.Stdout,
-		Stderr: os.Stderr,
+		Path:        path,
+		SysProcAttr: sysProcAttr(),
+		Args:        args,
+		Stdout:      os.Stdout,
+		Stderr:      os.Stderr,
 	}
 
 	err = cmd.Run()
diff --git a/cmd/oc-mirror/sysprocattr_linux.go b/cmd/oc-mirror/sysprocattr_linux.go
new file mode 100644
index 0000000000..351bfe9855
--- /dev/null
+++ b/cmd/oc-mirror/sysprocattr_linux.go
@@ -0,0 +1,11 @@
+package main
+
+import "syscall"
+
+func sysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		// Kill children if parent is dead
+		Pdeathsig: syscall.SIGKILL,
+		Setpgid:   true,
+	}
+}
diff --git a/cmd/oc-mirror/sysprocattr_other.go b/cmd/oc-mirror/sysprocattr_other.go
new file mode 100644
index 0000000000..4f9107d3ce
--- /dev/null
+++ b/cmd/oc-mirror/sysprocattr_other.go
@@ -0,0 +1,14 @@
+//go:build !linux
+
+package main
+
+import "syscall"
+
+// sysProcAttr creates a non-linux compatible SysProcAttr, primarily removing the use of
+// Pdeathsig which is only available on Linux. We Setpgid as true to tie the child process
+// to the parent process as a workaround.
+func sysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Setpgid: true,
+	}
+}
