diff --git a/requirements.cachito b/requirements.cachito
index 5313fcc..c6a012f 100644
--- a/requirements.cachito
+++ b/requirements.cachito
@@ -11,7 +11,7 @@
 # Note: this will fail CI checks (check-requirements.sh) and the downstream
 # Cachito build pipeline, so revert before submitting a PR.
 
-ironic-python-agent @ git+https://github.com/openshift/openstack-ironic-python-agent@18793a2971904c6f629df7f739f7a6b7215ea7a2
+ironic-python-agent @ git+https://github.com/openshift/openstack-ironic-python-agent@fb07ab65eb03af2831393d8e13f97412bd18f9c6
 
 # DIRECT DEPENDENCIES
 bcrypt===4.3.0
