diff --git a/pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go b/pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go
index 3c171548ee6e..521fdc1e7216 100644
--- a/pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go
+++ b/pkg/monitortestlibrary/disruptionlibrary/disruption_invariant_adapter.go
@@ -158,6 +158,11 @@ func createDisruptionJunit(
 		secondDetails, percentDetails = "added an additional 10s of grace for single node cluster", "added an additional 40% of grace for single node cluster"
 	}
 
+	if jobType.Topology == "dual" {
+		graceSeconds, gracePercent = 60, 1.4
+		secondDetails, percentDetails = "added an additional 60s of grace for dual replica cluster", "added an additional 40% of grace for dual replica cluster"
+	}
+
 	allowedSecsPlusGrace := allowedSecs + graceSeconds
 	allowedSecsPlusPercent := allowedSecs * gracePercent
 
diff --git a/pkg/monitortestlibrary/platformidentification/types.go b/pkg/monitortestlibrary/platformidentification/types.go
index fe5b68109a9d..ed7233db8e93 100644
--- a/pkg/monitortestlibrary/platformidentification/types.go
+++ b/pkg/monitortestlibrary/platformidentification/types.go
@@ -302,6 +302,8 @@ func GetJobType(ctx context.Context, clientConfig *rest.Config) (*JobType, error
 		topology = "single"
 	case configv1.ExternalTopologyMode:
 		topology = "external"
+	case configv1.DualReplicaTopologyMode:
+		topology = "dual"
 	}
 
 	return &JobType{
