diff --git a/images/router/haproxy/Dockerfile.ocp b/images/router/haproxy/Dockerfile.ocp
index c7cef4fe9..75436e69f 100644
--- a/images/router/haproxy/Dockerfile.ocp
+++ b/images/router/haproxy/Dockerfile.ocp
@@ -1,5 +1,5 @@
 FROM registry.ci.openshift.org/ocp/5.0:haproxy-router-base
-RUN INSTALL_PKGS="socat haproxy28 rsyslog procps-ng util-linux" && \
+RUN INSTALL_PKGS="socat haproxy32 rsyslog procps-ng util-linux" && \
     yum install -y --setopt=install_weak_deps=0 $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
