diff --git a/internal/operators/openshiftai/openshift_ai_manifests_test.go b/internal/operators/openshiftai/openshift_ai_manifests_test.go
index 36908f9829cd..d8092b634420 100644
--- a/internal/operators/openshiftai/openshift_ai_manifests_test.go
+++ b/internal/operators/openshiftai/openshift_ai_manifests_test.go
@@ -96,7 +96,7 @@ var _ = Describe("Manifest generation", func() {
 		Expect(clusters).To(HaveLen(1))
 	})
 
-	It("Includes the NVIDIA GPU accelerator profile if the NVIDIA GPU operator is enabled", func() {
+	It("Includes the NVIDIA GPU hardware profile if the NVIDIA GPU operator is enabled", func() {
 		// Enable the NVIDIA GPU operator:
 		cluster.MonitoredOperators = []*models.MonitoredOperator{
 			{
@@ -109,10 +109,10 @@ var _ = Describe("Manifest generation", func() {
 		Expect(err).ToNot(HaveOccurred())
 		objects := decodeManifests(manifests)
 
-		// Try to find the accelerator profile:
+		// Try to find the hardware profile:
 		var profiles []any
 		err = jqTool.Evaluate(
-			`.[] | select(.kind == "AcceleratorProfile") | .metadata.name`,
+			`.[] | select(.kind == "HardwareProfile") | .metadata.name`,
 			objects,
 			&profiles,
 		)
@@ -120,16 +120,16 @@ var _ = Describe("Manifest generation", func() {
 		Expect(profiles).To(ContainElement("nvidia-gpu"))
 	})
 
-	It("Doesn't include the NVIDIA GPU accelerator profile if the NVIDIA GPU operator isn't enabled", func() {
+	It("Doesn't include the NVIDIA GPU hardware profile if the NVIDIA GPU operator isn't enabled", func() {
 		// Convert the manifests into a list of objects:
 		_, manifests, err := operator.GenerateManifests(cluster)
 		Expect(err).ToNot(HaveOccurred())
 		objects := decodeManifests(manifests)
 
-		// Try to find the accelerator profile:
+		// Try to find the hardware profile:
 		var names []any
 		err = jqTool.Evaluate(
-			`.[] | select(.kind == "AcceleratorProfile") | .metadata.name`,
+			`.[] | select(.kind == "HardwareProfile") | .metadata.name`,
 			objects,
 			&names,
 		)
@@ -137,7 +137,7 @@ var _ = Describe("Manifest generation", func() {
 		Expect(names).ToNot(ContainElement("nvidia-gpu"))
 	})
 
-	It("Includes the AMD GPU accelerator profile if the AMD GPU operator is enabled", func() {
+	It("Includes the AMD GPU hardware profile if the AMD GPU operator is enabled", func() {
 		// Enable the AMD GPU operator:
 		cluster.MonitoredOperators = []*models.MonitoredOperator{
 			{
@@ -150,10 +150,10 @@ var _ = Describe("Manifest generation", func() {
 		Expect(err).ToNot(HaveOccurred())
 		objects := decodeManifests(manifests)
 
-		// Try to find the accelerator profile:
+		// Try to find the hardware profile:
 		var names []string
 		err = jqTool.Evaluate(
-			`.[] | select(.kind == "AcceleratorProfile") | .metadata.name`,
+			`.[] | select(.kind == "HardwareProfile") | .metadata.name`,
 			objects,
 			&names,
 		)
@@ -161,16 +161,16 @@ var _ = Describe("Manifest generation", func() {
 		Expect(names).To(ContainElement("amd-gpu"))
 	})
 
-	It("Doesn't include the AMD GPU accelerator profile if the AMD GPU operator isn't enabled", func() {
+	It("Doesn't include the AMD GPU hardware profile if the AMD GPU operator isn't enabled", func() {
 		// Convert the manifests into a list of objects:
 		_, manifests, err := operator.GenerateManifests(cluster)
 		Expect(err).ToNot(HaveOccurred())
 		objects := decodeManifests(manifests)
 
-		// Try to find the accelerator profile:
+		// Try to find the hardware profile:
 		var names []string
 		err = jqTool.Evaluate(
-			`.[] | select(.kind == "AcceleratorProfile") | .metadata.name`,
+			`.[] | select(.kind == "HardwareProfile") | .metadata.name`,
 			objects,
 			&names,
 		)
@@ -178,8 +178,8 @@ var _ = Describe("Manifest generation", func() {
 		Expect(names).ToNot(ContainElement("amd-gpu"))
 	})
 
-	It("Includes all accelerator profiles if all GPU operators are enabled", func() {
-		// Enable the AMD GPU operator:
+	It("Includes all hardware profiles if all GPU operators are enabled", func() {
+		// Enable AMD and NVIDIA GPU operators:
 		cluster.MonitoredOperators = []*models.MonitoredOperator{
 			{
 				Name: nvidiagpu.Operator.Name,
@@ -193,10 +193,10 @@ var _ = Describe("Manifest generation", func() {
 		Expect(err).ToNot(HaveOccurred())
 		objects := decodeManifests(manifests)
 
-		// Try to find the accelerator profile:
+		// Try to find the hardware profile:
 		var names []string
 		err = jqTool.Evaluate(
-			`.[] | select(.kind == "AcceleratorProfile") | .metadata.name`,
+			`.[] | select(.kind == "HardwareProfile") | .metadata.name`,
 			objects,
 			&names,
 		)
diff --git a/internal/operators/openshiftai/templates/custom/amd_acceleratorprofile.yaml b/internal/operators/openshiftai/templates/custom/amd_acceleratorprofile.yaml
deleted file mode 100644
index 98dd7aab446e..000000000000
--- a/internal/operators/openshiftai/templates/custom/amd_acceleratorprofile.yaml
+++ /dev/null
@@ -1,19 +0,0 @@
-{{ if .Cluster }}
-{{ range .Cluster.MonitoredOperators }}
-{{ if eq .Name "amd-gpu" }}
-apiVersion: dashboard.opendatahub.io/v1
-kind: AcceleratorProfile
-metadata:
-  namespace: redhat-ods-applications
-  name: amd-gpu
-spec:
-  displayName: AMD GPU
-  enabled: true
-  identifier: amd.com/gpu
-  tolerations:
-  - key: amd.com/gpu
-    operator: Exists
-    effect: NoSchedule
-{{ end }}
-{{ end }}
-{{ end }}
\ No newline at end of file
diff --git a/internal/operators/openshiftai/templates/custom/amd_hardwareprofile.yaml b/internal/operators/openshiftai/templates/custom/amd_hardwareprofile.yaml
new file mode 100644
index 000000000000..0821c909fdb2
--- /dev/null
+++ b/internal/operators/openshiftai/templates/custom/amd_hardwareprofile.yaml
@@ -0,0 +1,25 @@
+{{ if .Cluster }}
+{{ range .Cluster.MonitoredOperators }}
+{{ if eq .Name "amd-gpu" }}
+apiVersion: infrastructure.opendatahub.io/v1alpha1
+kind: HardwareProfile
+metadata:
+  namespace: redhat-ods-applications
+  name: amd-gpu
+spec:
+  identifiers:
+  - identifier: amd.com/gpu
+    displayName: AMD GPU
+    resourceType: Accelerator
+    defaultCount: 1
+    minCount: 0
+  scheduling:
+    type: Node
+    node:
+      tolerations:
+      - key: amd.com/gpu
+        operator: Exists
+        effect: NoSchedule
+{{ end }}
+{{ end }}
+{{ end }}
\ No newline at end of file
diff --git a/internal/operators/openshiftai/templates/custom/nvidia_acceleratorprofile.yaml b/internal/operators/openshiftai/templates/custom/nvidia_acceleratorprofile.yaml
deleted file mode 100644
index cfaa38ae7ad2..000000000000
--- a/internal/operators/openshiftai/templates/custom/nvidia_acceleratorprofile.yaml
+++ /dev/null
@@ -1,19 +0,0 @@
-{{ if .Cluster }}
-{{ range .Cluster.MonitoredOperators }}
-{{ if eq .Name "nvidia-gpu" }}
-apiVersion: dashboard.opendatahub.io/v1
-kind: AcceleratorProfile
-metadata:
-  namespace: redhat-ods-applications
-  name: nvidia-gpu
-spec:
-  displayName: NVIDIA GPU
-  enabled: true
-  identifier: nvidia.com/gpu
-  tolerations:
-  - key: nvidia.com/gpu
-    operator: Exists
-    effect: NoSchedule
-{{ end }}
-{{ end }}
-{{ end }}
\ No newline at end of file
diff --git a/internal/operators/openshiftai/templates/custom/nvidia_hardwareprofile.yaml b/internal/operators/openshiftai/templates/custom/nvidia_hardwareprofile.yaml
new file mode 100644
index 000000000000..0100a10a62dd
--- /dev/null
+++ b/internal/operators/openshiftai/templates/custom/nvidia_hardwareprofile.yaml
@@ -0,0 +1,25 @@
+{{ if .Cluster }}
+{{ range .Cluster.MonitoredOperators }}
+{{ if eq .Name "nvidia-gpu" }}
+apiVersion: infrastructure.opendatahub.io/v1alpha1
+kind: HardwareProfile
+metadata:
+  namespace: redhat-ods-applications
+  name: nvidia-gpu
+spec:
+  identifiers:
+  - identifier: nvidia.com/gpu
+    displayName: NVIDIA GPU
+    resourceType: Accelerator
+    defaultCount: 1
+    minCount: 0
+  scheduling:
+    type: Node
+    node:
+      tolerations:
+      - key: nvidia.com/gpu
+        operator: Exists
+        effect: NoSchedule
+{{ end }}
+{{ end }}
+{{ end }}
\ No newline at end of file
