diff --git a/hack/Dockerfile.debug b/hack/Dockerfile.debug
index 83a729a35..3b305d318 100644
--- a/hack/Dockerfile.debug
+++ b/hack/Dockerfile.debug
@@ -2,7 +2,7 @@ FROM registry.access.redhat.com/ubi9/ubi
 RUN yum install -y https://github.com/frobware/haproxy-builds/raw/master/rhaos-4.16-rhel-9/haproxy28-2.8.5-1.rhaos4.16.el9.x86_64.rpm
 RUN haproxy -vv
 RUN INSTALL_PKGS="rsyslog procps-ng util-linux socat" && \
-    yum install -y $INSTALL_PKGS && \
+    yum install -y --setopt=install_weak_deps=0 $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
     mkdir -p /var/lib/haproxy/router/{certs,cacerts,allowlists} && \
diff --git a/images/router/haproxy/Dockerfile b/images/router/haproxy/Dockerfile
index c254d75d9..77cbbf4de 100644
--- a/images/router/haproxy/Dockerfile
+++ b/images/router/haproxy/Dockerfile
@@ -1,6 +1,6 @@
 FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base-router
 RUN INSTALL_PKGS="socat haproxy28 rsyslog sysvinit-tools" && \
-    yum install -y $INSTALL_PKGS && \
+    yum install -y --setopt=install_weak_deps=0 $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
     mkdir -p /var/lib/haproxy/router/{certs,cacerts,allowlists} && \
diff --git a/images/router/haproxy/Dockerfile.ocp b/images/router/haproxy/Dockerfile.ocp
index 19051c749..78c1d20cf 100644
--- a/images/router/haproxy/Dockerfile.ocp
+++ b/images/router/haproxy/Dockerfile.ocp
@@ -1,6 +1,6 @@
 FROM registry.ci.openshift.org/ocp/4.21:haproxy-router-base
 RUN INSTALL_PKGS="socat haproxy28 rsyslog procps-ng util-linux" && \
-    yum install -y $INSTALL_PKGS && \
+    yum install -y --setopt=install_weak_deps=0 $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
     mkdir -p /var/lib/haproxy/router/{certs,cacerts,allowlists} && \
diff --git a/images/router/haproxy/Dockerfile.rhel b/images/router/haproxy/Dockerfile.rhel
index f91324fea..f5c508e8f 100644
--- a/images/router/haproxy/Dockerfile.rhel
+++ b/images/router/haproxy/Dockerfile.rhel
@@ -1,6 +1,6 @@
 FROM registry.svc.ci.openshift.org/ocp/4.0:base-router
 RUN INSTALL_PKGS="socat haproxy28 rsyslog sysvinit-tools" && \
-    yum install -y $INSTALL_PKGS && \
+    yum install -y --setopt=install_weak_deps=0 $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
     mkdir -p /var/lib/haproxy/router/{certs,cacerts,allowlists} && \
