diff --git a/web/cypress/e2e/coo/02.acm_alerting_ui.cy.ts b/web/cypress/e2e/coo/02.acm_alerting_ui.cy.ts
index a2f1c79d7..6b15968cd 100644
--- a/web/cypress/e2e/coo/02.acm_alerting_ui.cy.ts
+++ b/web/cypress/e2e/coo/02.acm_alerting_ui.cy.ts
@@ -27,6 +27,7 @@ describe('ACM Alerting UI', { tags: ['@coo', '@alerts'] }, () => {
 
   it('Navigate to Fleet Management > local-cluster > Observe > Alerting', () => {
     // check monitoring-plugin UI is not been affected
+    cy.switchPerspective('Core platform');
     nav.sidenav.clickNavLink(['Observe', 'Alerting']);
     commonPages.titleShouldHaveText('Alerting')
     nav.sidenav.clickNavLink(['Observe', 'Metrics']);
diff --git a/web/cypress/support/commands/coo-install-commands.ts b/web/cypress/support/commands/coo-install-commands.ts
index 9c989207a..5a772a9de 100644
--- a/web/cypress/support/commands/coo-install-commands.ts
+++ b/web/cypress/support/commands/coo-install-commands.ts
@@ -114,6 +114,7 @@ export const cooInstallUtils = {
       });
 
     if (Cypress.env('COO_UI_INSTALL')) {
+      cy.switchPerspective('Core platform');
       cy.get('#page-sidebar').then(($sidebar) => {
         const section = $sidebar.text().includes('Ecosystem') ? 'Ecosystem' : 'Operators';
         nav.sidenav.clickNavLink([section, 'Installed Operators']);
