diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index 284a910090..a3628cf240 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.24-openshift-4.22
+  tag: rhel-9-release-golang-1.25-openshift-4.22
diff --git a/images/manila-csi-plugin/Dockerfile b/images/manila-csi-plugin/Dockerfile
index c5cc7ac755..17e4b31bd4 100644
--- a/images/manila-csi-plugin/Dockerfile
+++ b/images/manila-csi-plugin/Dockerfile
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.22
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22
 WORKDIR /go/src/k8s.io/cloud-provider-openstack
 COPY . .
 RUN go build -o manila-csi-plugin ./cmd/manila-csi-plugin
