diff --git a/Dockerfile.rhel10 b/Dockerfile.rhel10
index 6584cec..dc78bda 100644
--- a/Dockerfile.rhel10
+++ b/Dockerfile.rhel10
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
+FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
 ARG KERNEL_VERSION=''
 ARG RT_KERNEL_VERSION=''
 ARG RHEL_VERSION=''
@@ -58,11 +58,11 @@ RUN if [ $(arch) == "x86_64" ] || [ $(arch) == "aarch64" ]; then \
 
 RUN dnf clean all && rm -rf /var/cache/dnf/*
 
-COPY manifests/01-openshift-imagestream.yaml /manifests/01-openshift-imagestream.yaml
+COPY manifests/01-openshift-imagestream-rhel10.yaml /manifests/01-openshift-imagestream.yaml
 COPY manifests/image-references-rhel10 /manifests/image-references
 
 LABEL io.k8s.description="driver-toolkit is a container with the kernel packages necessary for building driver containers for deploying kernel modules/drivers on OpenShift" \
-      name="driver-toolkit" \
+      name="driver-toolkit-10" \
       io.openshift.release.operator=true \
       version="0.1" \
       io.openshift.os.streamclass=rhel-10 \
diff --git a/manifests/01-openshift-imagestream-rhel10.yaml b/manifests/01-openshift-imagestream-rhel10.yaml
new file mode 100644
index 0000000..65d557a
--- /dev/null
+++ b/manifests/01-openshift-imagestream-rhel10.yaml
@@ -0,0 +1,26 @@
+kind: ImageStream
+apiVersion: image.openshift.io/v1
+metadata:
+  namespace: openshift
+  name: driver-toolkit
+  annotations:
+    include.release.openshift.io/hypershift: "true"
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  tags:
+  - name: latest-rhel-10
+    importPolicy:
+      scheduled: true
+      importMode: PreserveOriginal
+    from:
+      kind: DockerImage
+      name: example.com/image-reference-placeholder:driver-toolkit-10
+  - name: 0.0.1-snapshot-machine-os
+    importPolicy:
+      scheduled: true
+      importMode: PreserveOriginal
+    from:
+      kind: DockerImage
+      name: example.com/image-reference-placeholder:driver-toolkit-10
diff --git a/manifests/01-openshift-imagestream.yaml b/manifests/01-openshift-imagestream.yaml
index b10422b..e44558e 100644
--- a/manifests/01-openshift-imagestream.yaml
+++ b/manifests/01-openshift-imagestream.yaml
@@ -17,7 +17,7 @@ spec:
     from:
       kind: DockerImage
       name: example.com/image-reference-placeholder:driver-toolkit
-  - name: 0.0.1-snapshot-machine-os 
+  - name: 0.0.1-snapshot-machine-os
     importPolicy:
       scheduled: true
       importMode: PreserveOriginal
diff --git a/manifests/image-references-rhel10 b/manifests/image-references-rhel10
index 69f0f49..8524d60 100644
--- a/manifests/image-references-rhel10
+++ b/manifests/image-references-rhel10
@@ -2,10 +2,10 @@ kind: ImageStream
 apiVersion: image.openshift.io/v1
 spec:
   tags:
-  - name: driver-toolkit
+  - name: driver-toolkit-10
     from:
       kind: DockerImage
-      name: example.com/image-reference-placeholder:driver-toolkit
+      name: example.com/image-reference-placeholder:driver-toolkit-10
   - name: rhel-coreos-10
     from:
       kind: DockerImage
