diff --git a/OWNERS b/OWNERS
index 30d03fb18f..74e3aeb9fd 100644
--- a/OWNERS
+++ b/OWNERS
@@ -3,7 +3,6 @@ component: "Monitoring"
 reviewers:
 - simonpasquier
 - slashpai
-- raptorsun
 - jan--f
 - marioferh
 - danielmellado
@@ -13,7 +12,6 @@ reviewers:
 approvers:
 - simonpasquier
 - slashpai
-- raptorsun
 - jan--f
 - marioferh
 - danielmellado
@@ -21,6 +19,7 @@ approvers:
 - machine424
 
 emeritus_approvers:
+- raptorsun
 - dgrisonnet
 - paulfantom
 - brancz
