diff --git a/requirements.cachito b/requirements.cachito
index d309b91..a04820c 100644
--- a/requirements.cachito
+++ b/requirements.cachito
@@ -1,3 +1,3 @@
-ironic-python-agent @ git+https://github.com/openshift/openstack-ironic-python-agent@3c600e2da58c8ea8167a7885cc7e9a22c769d7e3
+ironic-python-agent @ git+https://github.com/openshift/openstack-ironic-python-agent@a123f29dfea2835cd184a48bbb9c48c2a8dfb4f0
 
 # DEPENDENCIES
