diff --git a/frontend/public/components/storage/shared.ts b/frontend/public/components/storage/shared.ts
index 64c8ee42ace..949b8f0fa18 100644
--- a/frontend/public/components/storage/shared.ts
+++ b/frontend/public/components/storage/shared.ts
@@ -121,6 +121,10 @@ export const provisionerAccessModeMapping: ProvisionerAccessModeMapping = Object
     Filesystem: ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany', 'ReadWriteOncePod'],
     Block: ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany', 'ReadWriteOncePod'],
   },
+  'nfs.csi.ceph.com': {
+    Filesystem: ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany', 'ReadWriteOncePod'],
+    partialMatch: true,
+  },
 });
 
 export const getAccessModeOptions = () => [
