diff --git a/pkg/operator/starter.go b/pkg/operator/starter.go
index 50b5e51a0..f0831b4a7 100644
--- a/pkg/operator/starter.go
+++ b/pkg/operator/starter.go
@@ -318,13 +318,20 @@ func newDegradedInertia() status.Inertia {
 		2*time.Minute,
 		// Nodes may temporarily become unready during upgrades while the machine/kubelet restarts.
 		// Use a longer inertia to avoid flapping the ClusterOperator Degraded condition.
-		status.InertiaCondition{
-			ConditionTypeMatcher: regexp.MustCompile("^" + condition.NodeControllerDegradedConditionType + "$"),
-			Duration:             10 * time.Minute,
-		},
+		inertiaForCondition(condition.NodeControllerDegradedConditionType, 10*time.Minute),
+		// Similarly, applying static pods to nodes that are being restarted may temporarily fail.
+		// Use a longer inertia to avoid flapping the ClusterOperator Degraded condition.
+		inertiaForCondition(condition.StaticPodsDegradedConditionType, 10*time.Minute),
 	).Inertia
 }
 
+func inertiaForCondition(cond string, duration time.Duration) status.InertiaCondition {
+	return status.InertiaCondition{
+		ConditionTypeMatcher: regexp.MustCompile("^" + cond + "$"),
+		Duration:             duration,
+	}
+}
+
 // deploymentConfigMaps is a list of configmaps that are directly copied for the current values.  A different actor/controller modifies these.
 // the first element should be the configmap that contains the static pod manifest
 var deploymentConfigMaps = []revision.RevisionResource{
diff --git a/pkg/operator/starter_test.go b/pkg/operator/starter_test.go
index 674ec2aa6..7b70a9094 100644
--- a/pkg/operator/starter_test.go
+++ b/pkg/operator/starter_test.go
@@ -19,6 +19,10 @@ func TestNewDegradedInertia(t *testing.T) {
 			conditionType: condition.NodeControllerDegradedConditionType,
 			expected:      10 * time.Minute,
 		},
+		{
+			conditionType: condition.StaticPodsDegradedConditionType,
+			expected:      10 * time.Minute,
+		},
 		{
 			conditionType: "TargetConfigControllerDegraded",
 			expected:      2 * time.Minute,
