diff --git a/frontend/public/components/masthead/masthead-toolbar.tsx b/frontend/public/components/masthead/masthead-toolbar.tsx
index a31b6217e3c..bbb5a46b257 100644
--- a/frontend/public/components/masthead/masthead-toolbar.tsx
+++ b/frontend/public/components/masthead/masthead-toolbar.tsx
@@ -791,6 +791,7 @@ const MastheadToolbarContents: FC<MastheadToolbarContentsProps> = ({
                   onClick={drawerToggle}
                   variant={notificationBadgeVariant}
                   count={notificationCount || 0}
+                  icon={<RhUiNotificationIcon />}
                   data-quickstart-id="qs-masthead-notifications"
                   className="co-masthead-button"
                 />
@@ -841,10 +842,9 @@ const MastheadToolbarContents: FC<MastheadToolbarContentsProps> = ({
                 onClick={drawerToggle}
                 variant={notificationBadgeVariant}
                 count={notificationCount}
+                icon={<RhUiNotificationIcon />}
                 data-quickstart-id="qs-masthead-notifications"
-              >
-                <RhUiNotificationIcon />
-              </NotificationBadge>
+              />
             )}
             <ToolbarItem>{renderMenu(true)}</ToolbarItem>
           </ToolbarGroup>
