diff --git a/frontend/__mocks__/resizeObserver.js b/frontend/__mocks__/resizeObserver.js
new file mode 100644
index 00000000000..ce23e3963fe
--- /dev/null
+++ b/frontend/__mocks__/resizeObserver.js
@@ -0,0 +1,6 @@
+global.ResizeObserver = class {
+  constructor() {}
+  disconnect() {}
+  observe() {}
+  unobserve() {}
+};
diff --git a/frontend/package.json b/frontend/package.json
index b722eecee2d..a7fefc6c1fe 100644
--- a/frontend/package.json
+++ b/frontend/package.json
@@ -126,6 +126,7 @@
       "./__mocks__/localStorage.ts",
       "./__mocks__/matchMedia.js",
       "./__mocks__/mutationObserver.js",
+      "./__mocks__/resizeObserver.js",
       "./__mocks__/serverFlags.js",
       "./__mocks__/textEncoderDecoder.ts",
       "./__mocks__/webpack.ts",
diff --git a/frontend/packages/console-shared/src/components/form-utils/__tests__/FormFooter.spec.tsx b/frontend/packages/console-shared/src/components/form-utils/__tests__/FormFooter.spec.tsx
index 3693516f451..5738fceca18 100644
--- a/frontend/packages/console-shared/src/components/form-utils/__tests__/FormFooter.spec.tsx
+++ b/frontend/packages/console-shared/src/components/form-utils/__tests__/FormFooter.spec.tsx
@@ -4,15 +4,6 @@ import { renderWithProviders } from '../../../test-utils/unit-test-utils';
 import type { FormFooterProps } from '../form-utils-types';
 import { FormFooter } from '../FormFooter';
 
-// Mock ResizeObserver
-global.ResizeObserver = class ResizeObserver {
-  observe = () => {};
-
-  unobserve = () => {};
-
-  disconnect = () => {};
-};
-
 describe('FormFooter', () => {
   let props: FormFooterProps;
 
diff --git a/frontend/packages/dev-console/src/components/deployments/__tests__/DeploymentForm.spec.tsx b/frontend/packages/dev-console/src/components/deployments/__tests__/DeploymentForm.spec.tsx
index 501394b8c18..bf325c4ee41 100644
--- a/frontend/packages/dev-console/src/components/deployments/__tests__/DeploymentForm.spec.tsx
+++ b/frontend/packages/dev-console/src/components/deployments/__tests__/DeploymentForm.spec.tsx
@@ -10,22 +10,6 @@ import MockForm from '../__mocks__/MockForm';
 import ContainerField from '../ContainerField';
 import DeploymentForm from '../DeploymentForm';
 
-class ResizeObserver {
-  observe() {
-    // do nothing
-  }
-
-  unobserve() {
-    // do nothing
-  }
-
-  disconnect() {
-    // do nothing
-  }
-}
-
-window.ResizeObserver = ResizeObserver;
-
 const mockContainerField: FC = () => {
   return <div>Container: xyz</div>;
 };
diff --git a/frontend/packages/dev-console/src/components/edit-application/__tests__/EditApplicationForm.spec.tsx b/frontend/packages/dev-console/src/components/edit-application/__tests__/EditApplicationForm.spec.tsx
index e09a899a050..dfcdb6c13f3 100644
--- a/frontend/packages/dev-console/src/components/edit-application/__tests__/EditApplicationForm.spec.tsx
+++ b/frontend/packages/dev-console/src/components/edit-application/__tests__/EditApplicationForm.spec.tsx
@@ -4,14 +4,6 @@ import { renderWithProviders } from '@console/shared/src/test-utils/unit-test-ut
 import { ApplicationFlowType } from '../edit-application-utils';
 import EditApplicationForm from '../EditApplicationForm';
 
-global.ResizeObserver = class ResizeObserver {
-  observe = () => {};
-
-  unobserve = () => {};
-
-  disconnect = () => {};
-};
-
 jest.mock('react-i18next', () => ({
   __esModule: true,
   useTranslation: () => ({
diff --git a/frontend/packages/dev-console/src/components/health-checks/__tests__/AddHealthChecks.spec.tsx b/frontend/packages/dev-console/src/components/health-checks/__tests__/AddHealthChecks.spec.tsx
index 3f700b75ffd..b4a95101b3a 100644
--- a/frontend/packages/dev-console/src/components/health-checks/__tests__/AddHealthChecks.spec.tsx
+++ b/frontend/packages/dev-console/src/components/health-checks/__tests__/AddHealthChecks.spec.tsx
@@ -6,14 +6,6 @@ import { getResourcesType } from '../../edit-application/edit-application-utils'
 import AddHealthChecks from '../AddHealthChecks';
 import { getHealthChecksData } from '../create-health-checks-probe-utils';
 
-global.ResizeObserver = class ResizeObserver {
-  observe = () => {};
-
-  unobserve = () => {};
-
-  disconnect = () => {};
-};
-
 jest.mock('react-i18next', () => ({
   __esModule: true,
   useTranslation: () => ({
diff --git a/frontend/packages/knative-plugin/src/components/add/__tests__/EventSource.spec.tsx b/frontend/packages/knative-plugin/src/components/add/__tests__/EventSource.spec.tsx
index a8f0c039087..2601b7a4645 100644
--- a/frontend/packages/knative-plugin/src/components/add/__tests__/EventSource.spec.tsx
+++ b/frontend/packages/knative-plugin/src/components/add/__tests__/EventSource.spec.tsx
@@ -5,14 +5,6 @@ import { EVENT_SOURCE_CONTAINER_KIND } from '../../../const';
 import { getEventSourceIcon } from '../../../utils/get-knative-icon';
 import { EventSource } from '../EventSource';
 
-global.ResizeObserver = class ResizeObserver {
-  observe = () => {};
-
-  unobserve = () => {};
-
-  disconnect = () => {};
-};
-
 jest.mock('@console/shared/src/components/formik-fields/SyncedEditorField', () => ({
   SyncedEditorField: ({ formContext }: { formContext?: { editor?: ReactNode } }) => (
     <>{formContext?.editor}</>
diff --git a/frontend/public/components/__tests__/command-line-tools.spec.tsx b/frontend/public/components/__tests__/command-line-tools.spec.tsx
index ce70d059dfe..98abecdde49 100644
--- a/frontend/public/components/__tests__/command-line-tools.spec.tsx
+++ b/frontend/public/components/__tests__/command-line-tools.spec.tsx
@@ -30,23 +30,7 @@ const obj = {
   loadError: null,
 };
 
-const nativeResizeObserver = window.ResizeObserver;
-class ResizeObserver {
-  observe() {}
-  unobserve() {}
-  disconnect() {}
-}
-
 describe('CommandLineTools', () => {
-  beforeAll(() => {
-    window.ResizeObserver = ResizeObserver;
-  });
-
-  afterAll(() => {
-    window.ResizeObserver = nativeResizeObserver;
-    jest.restoreAllMocks();
-  });
-
   describe('When ordering is correct', () => {
     it('shows oc CLI first in the displayed list', async () => {
       renderWithProviders(<CommandLineTools obj={obj} />);
