diff --git a/test/extended/node/node_e2e/image_registry_config.go b/test/extended/node/node_e2e/image_registry_config.go
index ce3a02189813..c9ec5a89fbf3 100644
--- a/test/extended/node/node_e2e/image_registry_config.go
+++ b/test/extended/node/node_e2e/image_registry_config.go
@@ -79,7 +79,7 @@ var _ = g.Describe("[Suite:openshift/disruptive-longrunning][sig-node][Disruptiv
 			}
 			imageConfig.Spec.RegistrySources.AllowedRegistries = []string{
 				"registry.access.redhat.com", "docker.io", "quay.io", searchRegistry,
-				"image-registry.openshift-image-registry.svc:5000",
+				"image-registry.openshift-image-registry.svc:5000", "quay-proxy.ci.openshift.org",
 			}
 			imageConfig.Spec.RegistrySources.ContainerRuntimeSearchRegistries = []string{
 				"registry.access.redhat.com", "docker.io", "quay.io", searchRegistry,
