diff --git a/Dockerfile.ocp b/Dockerfile.ocp
index 6d0da9f010..b86645196c 100644
--- a/Dockerfile.ocp
+++ b/Dockerfile.ocp
@@ -1,7 +1,7 @@
 FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 WORKDIR /go/src/k8s.io/kube-state-metrics
 COPY . .
-ENV GOFLAGS="-mod=vendor"
+ENV GOFLAGS="-mod=vendor -tags=timetzdata"
 RUN make build-local
 
 FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
