diff --git a/web/src/components/dashboards/legacy/useLegacyDashboardsProject.ts b/web/src/components/dashboards/legacy/useLegacyDashboardsProject.ts
index 008c6dba8..3e3c5949b 100644
--- a/web/src/components/dashboards/legacy/useLegacyDashboardsProject.ts
+++ b/web/src/components/dashboards/legacy/useLegacyDashboardsProject.ts
@@ -30,9 +30,6 @@ export const useLegacyDashboardsProject = () => {
         setOpenshiftProject(activeNamespace);
       }
     } else {
-      if (routeNamespace && activeNamespace !== routeNamespace) {
-        setActiveNamespace(routeNamespace);
-      }
       if (variableNamespace && variableNamespace !== routeNamespace) {
         dispatch(
           dashboardsPatchVariable('namespace', {
diff --git a/web/src/components/hooks/useMonitoringNamespace.ts b/web/src/components/hooks/useMonitoringNamespace.ts
index 38610015e..6509a0a93 100644
--- a/web/src/components/hooks/useMonitoringNamespace.ts
+++ b/web/src/components/hooks/useMonitoringNamespace.ts
@@ -1,7 +1,5 @@
 import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
-import { useEffect } from 'react';
 import { useParams } from 'react-router';
-import { useMonitoring } from '../../hooks/useMonitoring';
 
 /**
  * Utility hook to synchronize the namespace route in the URL with the activeNamespace
@@ -12,13 +10,6 @@ import { useMonitoring } from '../../hooks/useMonitoring';
 export const useMonitoringNamespace = () => {
   const { ns: routeNamespace } = useParams<{ ns?: string }>();
   const [activeNamespace, setActiveNamespace] = useActiveNamespace();
-  const { displayNamespaceSelector } = useMonitoring();
-
-  useEffect(() => {
-    if (routeNamespace && activeNamespace !== routeNamespace) {
-      setActiveNamespace(routeNamespace);
-    }
-  }, [routeNamespace, activeNamespace, setActiveNamespace, displayNamespaceSelector]);
 
   return {
     namespace: routeNamespace || activeNamespace,
