diff --git a/cmd/oadp/types.go b/cmd/oadp/types.go
index 0aaa7ce11d4..cd9ba7ab847 100644
--- a/cmd/oadp/types.go
+++ b/cmd/oadp/types.go
@@ -69,7 +69,7 @@ var (
 		"awsclusters.infrastructure.cluster.x-k8s.io", "awsmachinetemplates.infrastructure.cluster.x-k8s.io", "awsmachines.infrastructure.cluster.x-k8s.io",
 	}
 	agentResources = []string{
-		"agentclusters.infrastructure.cluster.x-k8s.io", "agentmachinetemplates.infrastructure.cluster.x-k8s.io", "agentmachines.infrastructure.cluster.x-k8s.io",
+		"agentclusters.capi-provider.agent-install.openshift.io", "agentmachinetemplates.capi-provider.agent-install.openshift.io", "agentmachines.capi-provider.agent-install.openshift.io",
 		"agents.agent-install.openshift.io", "infraenvs.agent-install.openshift.io", "nmstateconfigs.agent-install.openshift.io", "baremetalhosts.metal3.io",
 	}
 	kubevirtResources = []string{
