diff --git a/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx b/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
index 39b54303e4..c30d431602 100644
--- a/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
@@ -123,8 +123,17 @@ const ClusterDetailsForm = (props: ClusterDetailsFormProps) => {
         managedDomains,
         ocpVersions,
         urlSearchParams: search,
+        isSingleClusterFeatureEnabled,
       }),
-    [infraEnv, cluster, pullSecret, managedDomains, ocpVersions, search],
+    [
+      infraEnv,
+      cluster,
+      pullSecret,
+      managedDomains,
+      ocpVersions,
+      search,
+      isSingleClusterFeatureEnabled,
+    ],
   );
 
   const { t } = useTranslation();
diff --git a/libs/ui-lib/lib/ocm/services/ClusterDetailsService.ts b/libs/ui-lib/lib/ocm/services/ClusterDetailsService.ts
index dcef56f025..0b59743303 100644
--- a/libs/ui-lib/lib/ocm/services/ClusterDetailsService.ts
+++ b/libs/ui-lib/lib/ocm/services/ClusterDetailsService.ts
@@ -127,6 +127,7 @@ const ClusterDetailsService = {
     cluster,
     infraEnv,
     urlSearchParams,
+    isSingleClusterFeatureEnabled,
     ...args
   }: {
     cluster?: Cluster;
@@ -135,6 +136,7 @@ const ClusterDetailsService = {
     managedDomains: ManagedDomain[];
     ocpVersions: OpenshiftVersionOptionType[];
     urlSearchParams: string;
+    isSingleClusterFeatureEnabled?: boolean;
   }): OcmClusterDetailsValues {
     const values = getClusterDetailsInitialValues({
       cluster,
@@ -149,8 +151,12 @@ const ClusterDetailsService = {
       ? HostsNetworkConfigurationType.STATIC
       : HostsNetworkConfigurationType.DHCP;
 
+    const platform =
+      isSingleClusterFeatureEnabled && !cluster ? ('none' as PlatformType) : values.platform;
+
     return {
       ...values,
+      platform,
       cpuArchitecture,
       hostsNetworkConfigurationType,
       isCMNSupported: true,
