diff --git a/test/extended/networking/route_advertisements.go b/test/extended/networking/route_advertisements.go
index d82e0d5e3783..7f673f80f3ae 100644
--- a/test/extended/networking/route_advertisements.go
+++ b/test/extended/networking/route_advertisements.go
@@ -85,6 +85,7 @@ kind: NodeNetworkConfigurationPolicy
 metadata:
   name: extranet
 spec:
+  maxUnavailable: "100%"
   desiredState:
     interfaces:
     - name: enp3s0
@@ -97,6 +98,7 @@ kind: NodeNetworkConfigurationPolicy
 metadata:
   name: extranet
 spec:
+  maxUnavailable: "100%"
   desiredState:
     interfaces:
     - name: enp3s0
