diff --git a/OWNERS b/OWNERS
index cf2b72bd..20ae3447 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,19 +1,24 @@
 reviewers:
-- sjenning
-- enxebre
-- csrwng
-- muraee
 - bryan-cox
-- jparrill
+- clebs
+- csrwng
+- devguyio
 - dgrisonnet
-- swghosh
-approvers:
-- sjenning
 - enxebre
-- csrwng
+- jparrill
 - muraee
+- Nirshal
+- sdminonne
+- sjenning
+- swghosh
+approvers:
 - bryan-cox
-- jparrill
+- csrwng
+- devguyio
 - dgrisonnet
+- enxebre
+- jparrill
+- muraee
+- sjenning
 - swghosh
 component: hypershift
