diff --git a/test/e2e/util/fixture.go b/test/e2e/util/fixture.go
index 48d35eacec4..8e9ab9e123b 100644
--- a/test/e2e/util/fixture.go
+++ b/test/e2e/util/fixture.go
@@ -300,7 +300,7 @@ func validateAWSGuestResourcesDeletedFunc(ctx context.Context, t *testing.T, inf
 		var lastOutput *resourcegroupstaggingapi.GetResourcesOutput
 
 		// Find load balancers, persistent volumes, or s3 buckets belonging to the guest cluster
-		err := wait.PollUntilContextTimeout(ctx, 20*time.Second, 25*time.Minute, false, func(ctx context.Context) (bool, error) {
+		err := wait.PollUntilContextTimeout(ctx, 20*time.Second, 15*time.Minute, false, func(ctx context.Context) (bool, error) {
 			// Filter get cluster resources.
 			output, err := taggingClient.GetResources(ctx, &resourcegroupstaggingapi.GetResourcesInput{
 				ResourceTypeFilters: []string{
