diff --git a/test/extended/internalreleaseimage/helper.go b/test/extended/internalreleaseimage/helper.go
index 8747c9c76f50..b650a5f1344f 100644
--- a/test/extended/internalreleaseimage/helper.go
+++ b/test/extended/internalreleaseimage/helper.go
@@ -190,6 +190,12 @@ func (h *IRITestHelper) CreateSimpleNamespace() string {
 		o.Expect(err).NotTo(o.HaveOccurred(), "Timed out waiting for namespace %s to get SCC annotations", createdNs.Name)
 	}
 
+	err = exutil.WaitForServiceAccount(h.oc.AdminKubeClient().CoreV1().ServiceAccounts(createdNs.Name), "default")
+	if err != nil {
+		h.DeleteNamespace(createdNs.Name)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Timed out waiting for default ServiceAccount in namespace %s", createdNs.Name)
+	}
+
 	return createdNs.Name
 }
 
