diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index a3628cf..1d88a59 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.25-openshift-4.22
+  tag: rhel-9-release-golang-1.26-openshift-5.0
diff --git a/Dockerfile b/Dockerfile
index efd5ee7..4e9e3eb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,12 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 WORKDIR /go/src/github.com/openshift/network-tools
 COPY . .
 
 # needed for ovnkube-trace
-FROM registry.ci.openshift.org/ocp/4.22:ovn-kubernetes AS ovnkube-trace
+FROM registry.ci.openshift.org/ocp/5.0:ovn-kubernetes AS ovnkube-trace
 
 # tools (openshift-tools) is based off cli
-FROM registry.ci.openshift.org/ocp/4.22:tools
+FROM registry.ci.openshift.org/ocp/5.0:tools
 COPY --from=builder /go/src/github.com/openshift/network-tools/debug-scripts/ /opt/bin/
 COPY --from=ovnkube-trace /usr/bin/ovnkube-trace /usr/bin/
 
