diff --git a/pkg/resource/prunercronjob.go b/pkg/resource/prunercronjob.go
index 3c2776aedd..2fccc5bbcc 100644
--- a/pkg/resource/prunercronjob.go
+++ b/pkg/resource/prunercronjob.go
@@ -180,7 +180,7 @@ done
 	}
 	cj.Spec.JobTemplate.Labels = map[string]string{"created-by": gcj.GetName()}
 	cj.Spec.JobTemplate.Spec.Template.Labels = defaults.PrunerPodLabels
-	cj.Spec.JobTemplate.Annotations = map[string]string{securityv1.RequiredSCCAnnotation: "restricted-v2"}
+	cj.Spec.JobTemplate.Spec.Template.Annotations = map[string]string{securityv1.RequiredSCCAnnotation: "restricted-v2"}
 	return cj, nil
 }
 
