diff --git a/web/cypress/support/commands/utility-commands.ts b/web/cypress/support/commands/utility-commands.ts
index 15d2fdd21..dbbb58e80 100644
--- a/web/cypress/support/commands/utility-commands.ts
+++ b/web/cypress/support/commands/utility-commands.ts
@@ -135,7 +135,6 @@ Cypress.Commands.add('waitUntilWithCustomTimeout', (
       }
     });
     cy.get(`a[data-test^="${pod}"]`).eq(0).as('podLink').click();
-      cy.get('@podLink').should('be.visible').click();
       cy.byPFRole('rowgroup').find('td').eq(1).scrollIntoView().should('be.visible').then(($td) => {
         cy.log('Pod image: ' + $td.text());
       });
