diff --git a/pkg/cli/image/archive/archive.go b/pkg/cli/image/archive/archive.go
index 93c11c5c73..a6730df622 100644
--- a/pkg/cli/image/archive/archive.go
+++ b/pkg/cli/image/archive/archive.go
@@ -2,6 +2,7 @@ package archive
 
 import (
 	"archive/tar"
+	"errors"
 	"fmt"
 	"io"
 	"os"
@@ -402,12 +403,12 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 			continue
 		}
 		if err := system.Lsetxattr(path, key, []byte(value), 0); err != nil {
-			if err == system.ErrNotSupportedPlatform {
+			if errors.Is(err, system.ErrNotSupportedPlatform) {
 				// we ignore not supported platform errors
 				// to proceed archiving on platforms like darwin.
 				continue
 			}
-			if err == syscall.ENOTSUP {
+			if errors.Is(err, syscall.ENOTSUP) {
 				// We ignore errors here because not all graphdrivers support
 				// xattrs *cough* old versions of AUFS *cough*. However only
 				// ENOTSUP should be emitted in that case, otherwise we still
