diff --git a/pkg/cmd/openshift-tests/run-upgrade/flags.go b/pkg/cmd/openshift-tests/run-upgrade/flags.go
index 00be9050bf5a..05cb407db9f5 100644
--- a/pkg/cmd/openshift-tests/run-upgrade/flags.go
+++ b/pkg/cmd/openshift-tests/run-upgrade/flags.go
@@ -48,7 +48,7 @@ func NewRunUpgradeSuiteFlags(streams genericclioptions.IOStreams, fromRepository
 func (f *RunUpgradeSuiteFlags) BindFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&f.FromRepository, "from-repository", f.FromRepository, "A container image repository to retrieve test images from.")
 	flags.StringVar(&f.ProviderTypeOrJSON, "provider", f.ProviderTypeOrJSON, "The cluster infrastructure provider. Will automatically default to the correct value.")
-	flags.StringVar(&f.ToImage, "to-image", f.ToImage, "Specify the image to test an upgrade to.")
+	flags.StringVar(&f.ToImage, "to-image", f.ToImage, "Specify the images to test an upgrade to, separated by comma.")
 	flags.StringSliceVar(&f.TestOptions, "options", f.TestOptions, "A set of KEY=VALUE options to control the test. See the help text.")
 	f.GinkgoRunSuiteOptions.BindFlags(flags)
 	f.TestSuiteSelectionFlags.BindFlags(flags)
diff --git a/pkg/cmd/openshift-tests/run-upgrade/options.go b/pkg/cmd/openshift-tests/run-upgrade/options.go
index 2d72aeebc037..193e14402ea2 100644
--- a/pkg/cmd/openshift-tests/run-upgrade/options.go
+++ b/pkg/cmd/openshift-tests/run-upgrade/options.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/openshift/origin/pkg/monitortestframework"
 
@@ -109,10 +110,19 @@ func (o *RunUpgradeSuiteOptions) Run(ctx context.Context) error {
 		return err
 	}
 
+	last := o.ToImage
+	if strings.Contains(last, ",") {
+		splits := strings.Split(o.ToImage, ",")
+		last = strings.TrimSpace(splits[len(splits)-1])
+		if last == "" {
+			return fmt.Errorf("failed to get the final upgrade destination from %s", o.ToImage)
+		}
+	}
+
 	// TODO the gingkoRunSuiteOptions needs to have flags then calculated options to express specified versus computed values
 	monitorTestInfo := monitortestframework.MonitorTestInitializationInfo{
 		ClusterStabilityDuringTest:        monitortestframework.Stable,
-		UpgradeTargetPayloadImagePullSpec: o.ToImage,
+		UpgradeTargetPayloadImagePullSpec: last,
 		ExactMonitorTests:                 o.GinkgoRunSuiteOptions.ExactMonitorTests,
 		DisableMonitorTests:               o.GinkgoRunSuiteOptions.DisableMonitorTests,
 	}
