diff --git a/pkg/controller/config.go b/pkg/controller/config.go
index 29a5f34c9..c049cd31d 100644
--- a/pkg/controller/config.go
+++ b/pkg/controller/config.go
@@ -2,7 +2,7 @@ package controller
 
 import (
 	"encoding/json"
-	"io/ioutil"
+	"os"
 
 	kyaml "k8s.io/apimachinery/pkg/util/yaml"
 
@@ -28,7 +28,7 @@ func LoadConfig(cliConfig string) ClusterMachineApproverConfig {
 		return config
 	}
 
-	content, err := ioutil.ReadFile(cliConfig)
+	content, err := os.ReadFile(cliConfig)
 	if err != nil {
 		klog.Infof("using default as failed to load config %s: %v", cliConfig, err)
 		return config
