diff --git a/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts b/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
index eed39497a50..5753531da2a 100644
--- a/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
+++ b/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
@@ -99,6 +99,8 @@ describe('CustomResourceDefinitions', () => {
     listPage.isCreateButtonVisible();
     listPage.dvRows.shouldBeLoaded();
     listPage.dvRows.clickKebabAction(`CRD${testName}`, 'View instances');
+    cy.url().should('include', `/k8s/all-namespaces/test.example.com~v1~CRD${testName}`);
+    listPage.isCreateButtonVisible();
     listPage.clickCreateYAMLbutton();
     yamlEditor.isLoaded();
     yamlEditor.getEditorContent().then((content) => {
