diff --git a/frontend/packages/console-shared/src/components/heading/PageHeading.tsx b/frontend/packages/console-shared/src/components/heading/PageHeading.tsx
index 75e6a171057..c1cb154dffa 100644
--- a/frontend/packages/console-shared/src/components/heading/PageHeading.tsx
+++ b/frontend/packages/console-shared/src/components/heading/PageHeading.tsx
@@ -9,10 +9,8 @@ import { Breadcrumbs } from '@console/shared/src/components/breadcrumbs/Breadcru
 
 import './PageHeading.scss';
 
-interface PageHeadingLinkProps extends Omit<PageHeaderLinkProps, 'label'> {
+interface PageHeadingLinkProps extends PageHeaderLinkProps {
   'data-test'?: string;
-  /** Title for the link */
-  label: ReactNode | string;
 }
 
 export type PageHeadingProps = {
