diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index d85edef45a..2ecde889e7 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -11,14 +11,16 @@ aliases:
   - tssurya
   core-reviewers:
   - arghosh93
-  - arkadeepsen 
+  - arkadeepsen
   - bpickard22
   - danwinship
   - jcaamano
   - kyrtapz
-  - martinkennelly 
+  - martinkennelly
+  - marty-power
+  - mattedallo
   - miheer
   - pliurh
-  - pperiyasamy 
-  - ricky-rav
+  - pperiyasamy
+  - taanyas
   - tssurya
