diff --git a/OWNERS b/OWNERS
index 3a1d02735..53c0b12e3 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,12 +1,23 @@
 # See the OWNERS docs at https://go.k8s.io/owners
 
 reviewers:
-  - sjenning
-  - enxebre
+  - bryan-cox
+  - clebs
   - csrwng
+  - devguyio
+  - enxebre
+  - jparrill
+  - muraee
+  - Nirshal
   - relyt0925
-approvers:
+  - sdminonne
   - sjenning
-  - enxebre
+approvers:
+  - bryan-cox
   - csrwng
+  - devguyio
+  - enxebre
+  - jparrill
+  - muraee
+  - sjenning
 component: hypershift
