diff --git a/frontend/packages/topology/locales/en/topology.json b/frontend/packages/topology/locales/en/topology.json
index 51a919cd32a..391b6fc896e 100644
--- a/frontend/packages/topology/locales/en/topology.json
+++ b/frontend/packages/topology/locales/en/topology.json
@@ -74,6 +74,7 @@
   "Save": "Save",
   "Connect <1>{{sourceLabel}}</1> to": "Connect <1>{{sourceLabel}}</1> to",
   "Unable to move connector of type {{type}}.": "Unable to move connector of type {{type}}.",
+  "Drop file ({{fileTypes}}) here": "Drop file ({{fileTypes}}) here",
   "Loading is taking longer than expected": "Loading is taking longer than expected",
   "We noticed that it is taking a long time to visualize your application Topology. You can use Search to find specific resources or click Continue to keep waiting.": "We noticed that it is taking a long time to visualize your application Topology. You can use Search to find specific resources or click Continue to keep waiting.",
   "Go to Search": "Go to Search",
@@ -85,7 +86,6 @@
   "Graph view": "Graph view",
   "Shortcuts": "Shortcuts",
   "View shortcuts": "View shortcuts",
-  "Drop file ({{fileTypes}}) here": "Drop file ({{fileTypes}}) here",
   "Prerequisites": "Prerequisites",
   "Quick Starts": "Quick Starts",
   "Start": "Start",
diff --git a/frontend/packages/topology/src/components/page/DroppableTopologyComponent.tsx b/frontend/packages/topology/src/components/page/DroppableTopologyComponent.tsx
index 9c4da391760..f0bead542e4 100644
--- a/frontend/packages/topology/src/components/page/DroppableTopologyComponent.tsx
+++ b/frontend/packages/topology/src/components/page/DroppableTopologyComponent.tsx
@@ -1,56 +1,54 @@
-import { useContext } from 'react';
+import { useCallback, useContext } from 'react';
+import type { MultipleFileUploadProps } from '@patternfly/react-core';
+import { MultipleFileUpload } from '@patternfly/react-core';
 import type { Model } from '@patternfly/react-topology';
-import type { DropTargetConnector } from 'react-dnd';
-import { DropTarget } from 'react-dnd';
-import { NativeTypes } from 'react-dnd-html5-backend';
-import type { DropTargetMonitor } from 'react-dnd/lib/interfaces';
-import type { FileUploadContextType } from '@console/app/src/components/file-upload/file-upload-context';
+import { useTranslation } from 'react-i18next';
 import { FileUploadContext } from '@console/app/src/components/file-upload/file-upload-context';
-import withDragDropContext from '@console/internal/components/utils/drag-drop-context';
+import type { FileUploadContextType } from '@console/app/src/components/file-upload/file-upload-context';
 import type { TopologyViewType } from '../../topology-types';
-import type { TopologyViewProps } from './TopologyView';
 import TopologyView from './TopologyView';
 
-const boxTarget = {
-  drop(props, monitor) {
-    if (props.onDrop && monitor.isOver()) {
-      props.onDrop(monitor);
-    }
-  },
-};
-
-const DroppableTopology = DropTarget(
-  NativeTypes.FILE,
-  boxTarget,
-  (connectObj: DropTargetConnector, monitor: DropTargetMonitor, props: TopologyViewProps) => {
-    return {
-      connectDropTarget: connectObj.dropTarget(),
-      isOver: monitor.isOver(),
-      canDrop: monitor.canDrop() && props.canDropFile,
-    };
-  },
-)(TopologyView);
+export const DroppableTopologyComponent: React.FC<DroppableTopologyComponentProps> = (props) => {
+  const { t } = useTranslation('topology');
+  const { setFileUpload, extensions } = useContext<FileUploadContextType>(FileUploadContext);
+  const fileTypes = extensions.map((ex) => `.${ex}`).toString();
 
-export const DroppableTopologyComponent = withDragDropContext<DroppableTopologyComponentProps>(
-  (props) => {
-    const { setFileUpload, extensions } = useContext<FileUploadContextType>(FileUploadContext);
+  const canDropFile = extensions.length > 0;
 
-    const handleFileDrop = (monitor: DropTargetMonitor) => {
-      if (!monitor) {
-        return;
+  const handleFileDrop = useCallback<MultipleFileUploadProps['onFileDrop']>(
+    (_event, files) => {
+      const [file] = files;
+      if (file) {
+        setFileUpload(file);
       }
-      const [file] = monitor.getItem().files;
-      if (!file) {
-        return;
-      }
-      setFileUpload(file);
-    };
+    },
+    [setFileUpload],
+  );
+
+  if (!canDropFile) {
+    return <TopologyView {...props} />;
+  }
 
-    return (
-      <DroppableTopology {...props} onDrop={handleFileDrop} canDropFile={extensions.length > 0} />
-    );
-  },
-);
+  return (
+    <MultipleFileUpload
+      style={{ display: 'contents' }}
+      onFileDrop={handleFileDrop}
+      dropzoneProps={{
+        noClick: true,
+        noKeyboard: true,
+      }}
+    >
+      <div className="co-file-dropzone co-file-dropzone__flex">
+        <div className="co-file-dropzone-container">
+          <span className="co-file-dropzone__drop-text">
+            {t('Drop file ({{fileTypes}}) here', { fileTypes })}
+          </span>
+        </div>
+        <TopologyView {...props} />
+      </div>
+    </MultipleFileUpload>
+  );
+};
 
 export type DroppableTopologyComponentProps = {
   model: Model;
diff --git a/frontend/packages/topology/src/components/page/TopologyView.scss b/frontend/packages/topology/src/components/page/TopologyView.scss
index 3ed7a5d8e63..c55b6dea01e 100644
--- a/frontend/packages/topology/src/components/page/TopologyView.scss
+++ b/frontend/packages/topology/src/components/page/TopologyView.scss
@@ -39,17 +39,6 @@
     margin-right: var(--pf-t--global--spacer--2xl);
   }
 
-  &__dropzone {
-    background-color: rgba(0, 102, 204, 0.1) !important;
-    color: var(--pf-t--global--text--color--regular) !important;
-    margin: 0 !important;
-  }
-
-  &__dropzone-text {
-    top: 0%;
-    padding-top: var(--pf-t--global--spacer--lg);
-  }
-
   .pf-topology-container {
     display: flex;
   }
diff --git a/frontend/packages/topology/src/components/page/TopologyView.tsx b/frontend/packages/topology/src/components/page/TopologyView.tsx
index a08188a0cbb..44e3ba58e80 100644
--- a/frontend/packages/topology/src/components/page/TopologyView.tsx
+++ b/frontend/packages/topology/src/components/page/TopologyView.tsx
@@ -1,14 +1,9 @@
 import type { FC } from 'react';
 import { useContext, useState, useRef, useCallback, useMemo, useEffect } from 'react';
 import { Drawer, DrawerContent, DrawerContentBody, Stack, StackItem } from '@patternfly/react-core';
-import { css } from '@patternfly/react-styles';
 import type { GraphElement, Model, Visualization } from '@patternfly/react-topology';
 import { isGraph, TopologyQuadrant } from '@patternfly/react-topology';
-import type { ConnectDropTarget, DropTargetMonitor } from 'react-dnd';
-import { useTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
-import type { FileUploadContextType } from '@console/app/src/components/file-upload/file-upload-context';
-import { FileUploadContext } from '@console/app/src/components/file-upload/file-upload-context';
 import { useAddToProjectAccess } from '@console/dev-console/src/utils/useAddToProjectAccess';
 import { useResolvedExtensions } from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
 import type {
@@ -73,11 +68,6 @@ export interface TopologyViewProps {
   model: Model;
   namespace: string;
   viewType: TopologyViewType;
-  connectDropTarget?: ConnectDropTarget;
-  isOver?: boolean;
-  canDrop?: boolean;
-  onDrop?: (monitor: DropTargetMonitor) => void;
-  canDropFile?: boolean;
 }
 
 type ComponentProps = TopologyViewProps & StateProps & DispatchProps;
@@ -90,11 +80,7 @@ export const ConnectedTopologyView: FC<ComponentProps> = ({
   application,
   onSupportedFiltersChange,
   onSupportedKindsChange,
-  connectDropTarget,
-  isOver,
-  canDrop,
 }) => {
-  const { t } = useTranslation();
   const fireTelemetryEvent = useTelemetry();
   const { setTopologyFilters: onFiltersChange } = useContext(FilterContext);
   const { getQueryArgument, setQueryArgument, removeQueryArgument } = useQueryParamsMutator();
@@ -138,13 +124,9 @@ export const ConnectedTopologyView: FC<ComponentProps> = ({
   }>({});
   const [filtersLoaded, setFiltersLoaded] = useState<boolean>(false);
   const queryParams = useQueryParams();
-  const { extensions: supportedFileExtensions } = useContext<FileUploadContextType>(
-    FileUploadContext,
-  );
 
   const searchParams = queryParams.get(TOPOLOGY_SEARCH_FILTER_KEY);
   const labelParams = queryParams.get(TOPOLOGY_LABELS_FILTER_KEY);
-  const fileTypes = supportedFileExtensions.map((ex) => `.${ex}`).toString();
 
   const onSelect = useCallback(
     (entity?: GraphElement) => {
@@ -325,45 +307,28 @@ export const ConnectedTopologyView: FC<ComponentProps> = ({
           />
         </StackItem>
         <StackItem isFilled className="pf-topology-container">
-          <div className="co-file-dropzone co-file-dropzone__flex">
-            <Drawer isExpanded={isSidebarAvailable} isInline>
-              <DrawerContent
-                panelContent={
-                  <TopologySideBar onClose={() => onSelect()}>
-                    <SelectedEntityDetails selectedEntity={selectedEntity} />
-                  </TopologySideBar>
-                }
-              >
-                <DrawerContentBody>
-                  <div className="pf-topology-content ocs-quick-search-modal__no-backdrop">
-                    {canDrop && isOver && (
-                      <div
-                        className={css(
-                          'co-file-dropzone-container',
-                          'co-file-dropzone--drop-over',
-                          'odc-topology__dropzone',
-                        )}
-                      >
-                        <span className="co-file-dropzone__drop-text odc-topology__dropzone-text">
-                          {t('topology~Drop file ({{fileTypes}}) here', { fileTypes })}
-                        </span>
-                      </div>
-                    )}
-                    {nodesLength > MAX_NODES_LIMIT && !showTopologyAnyway ? (
-                      <LimitExceededState
-                        onShowTopologyAnyway={() => setShowTopologyAnyway(true)}
-                      />
-                    ) : (
-                      viewContent
-                    )}
-                    {!model.nodes?.length ? (
-                      <TopologyEmptyState setIsQuickSearchOpen={setIsQuickSearchOpenAndFireEvent} />
-                    ) : null}
-                  </div>
-                </DrawerContentBody>
-              </DrawerContent>
-            </Drawer>
-          </div>
+          <Drawer isExpanded={isSidebarAvailable} isInline>
+            <DrawerContent
+              panelContent={
+                <TopologySideBar onClose={() => onSelect()}>
+                  <SelectedEntityDetails selectedEntity={selectedEntity} />
+                </TopologySideBar>
+              }
+            >
+              <DrawerContentBody>
+                <div className="pf-topology-content ocs-quick-search-modal__no-backdrop">
+                  {nodesLength > MAX_NODES_LIMIT && !showTopologyAnyway ? (
+                    <LimitExceededState onShowTopologyAnyway={() => setShowTopologyAnyway(true)} />
+                  ) : (
+                    viewContent
+                  )}
+                  {!model.nodes?.length ? (
+                    <TopologyEmptyState setIsQuickSearchOpen={setIsQuickSearchOpenAndFireEvent} />
+                  ) : null}
+                </div>
+              </DrawerContentBody>
+            </DrawerContent>
+          </Drawer>
         </StackItem>
         <TopologyQuickSearch
           namespace={namespace}
@@ -374,9 +339,7 @@ export const ConnectedTopologyView: FC<ComponentProps> = ({
     </div>
   );
 
-  return typeof connectDropTarget === 'function'
-    ? connectDropTarget(topologyViewComponent)
-    : topologyViewComponent;
+  return topologyViewComponent;
 };
 
 const TopologyStateToProps = (state: RootState): StateProps => {
diff --git a/frontend/public/components/utils/_file-input.scss b/frontend/public/components/utils/_file-input.scss
index 82c1fb9f321..9e72a33152c 100644
--- a/frontend/public/components/utils/_file-input.scss
+++ b/frontend/public/components/utils/_file-input.scss
@@ -1,28 +1,7 @@
-.co-btn-file {
-  overflow: hidden;
-  position: relative;
-  input[type='file'] {
-    cursor: inherit;
-    display: block;
-    filter: alpha(opacity=0);
-    font-size: 100px;
-    min-height: 100%;
-    min-width: 100%;
-    opacity: 0;
-    outline: none;
-    position: absolute;
-    right: 0;
-    text-align: right;
-    top: 0;
-    z-index: 2;
-  }
-}
-
 .co-file-dropzone__textarea {
   min-height: 300px;
 }
 
-
 .co-file-dropzone {
   position: relative;
 
@@ -41,10 +20,6 @@
     right: 0;
     top: 0;
     z-index: var(--pf-t--global--z-index--2xl);
-    &.co-file-dropzone--drop-over {
-      color: var(--pf-t--global--color--brand--default);
-      outline-color: var(--pf-t--global--color--brand--default);
-    }
   }
 }
 
