diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index 166476c54caa..30f3c7c6ca7c 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -21,7 +21,6 @@ aliases:
   core-reviewers:
     - enxebre
     - csrwng
-    - sjenning
     - muraee
     - bryan-cox
     - cblecker
@@ -30,6 +29,7 @@ aliases:
     - sdminonne
     - clebs
     - Nirshal
+    - ironcladlou
   karpenter-approvers:
     - elmiko
     - enxebre
