diff --git a/Containerfile.control-plane b/Containerfile.control-plane
index 8fa33328164..ee5bd45d5da 100644
--- a/Containerfile.control-plane
+++ b/Containerfile.control-plane
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/go-toolset:1.25.3-1765311584 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.25.7-1773318690 AS builder
 
 WORKDIR /hypershift
 
@@ -7,7 +7,7 @@ COPY --chown=default . .
 RUN make control-plane-operator \
   && make control-plane-pki-operator
 
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.7-1773204619
 COPY --from=builder /hypershift/bin/control-plane-operator /usr/bin/control-plane-operator
 COPY --from=builder /hypershift/bin/control-plane-pki-operator /usr/bin/control-plane-pki-operator
 
diff --git a/Containerfile.operator b/Containerfile.operator
index 2f39da6a5fe..874871a6b55 100644
--- a/Containerfile.operator
+++ b/Containerfile.operator
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/go-toolset:1.25.3-1765311584 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.25.7-1773318690 AS builder
 
 WORKDIR /hypershift
 
@@ -10,7 +10,7 @@ RUN make hypershift \
   && make product-cli \
   && make karpenter-operator
 
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.7-1773204619
 COPY --from=builder /hypershift/bin/hypershift \
   /hypershift/bin/hypershift-no-cgo \
   /hypershift/bin/hcp \
