diff --git a/OWNERS b/OWNERS
index a5635860b..91fb7e431 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,7 +1,9 @@
 reviewers:
+- control-plane-approvers
 - flavianmissi
 - ricardomaraschini
 approvers:
+- control-plane-approvers
 - ricardomaraschini
 - flavianmissi
 component: "Image Registry"
diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
new file mode 100644
index 000000000..66464a003
--- /dev/null
+++ b/OWNERS_ALIASES
@@ -0,0 +1,16 @@
+aliases:
+  control-plane-approvers:
+    - ardaguclu
+    - atiratree
+    - benluddy
+    - bertinatto
+    - everettraven
+    - flavianmissi
+    - gangwgr
+    - ingvagabund
+    - kaleemsiddiqu
+    - p0lyn0mial
+    - rh-roman
+    - ricardomaraschini
+    - tjungblu
+    - xueqzhan
