diff --git a/Makefile b/Makefile
index d9a06aac6..c6a2d7f6a 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ FEATURES    ?=incidents,perses-dashboards,dev-config
 GOLANGCI_LINT = $(shell pwd)/_output/tools/bin/golangci-lint
 GOLANGCI_LINT_VERSION ?= v2.11.3
 
-export NODE_OPTIONS?=--max_old_space_size=4096
+export NODE_OPTIONS?=--max_old_space_size=8192
 
 .PHONY: install-frontend
 install-frontend:
