diff --git a/OWNERS b/OWNERS
index 310848d04a4..69bc7a8f734 100644
--- a/OWNERS
+++ b/OWNERS
@@ -61,6 +61,8 @@ filters:
     labels:
     - area/ai
   "^api/.*":
+    approvers:
+    - api-approvers
     labels:
     - area/api
   "^availability-prober/.*":
diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index 142dac1a900..d73ad570322 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -1,4 +1,10 @@
 aliases:
+  api-approvers:
+    - enxebre
+    - csrwng
+    - sjenning
+    - joelspeed
+    - everettraven
   tech-leads:
     - enxebre
     - csrwng
