diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index 284a9100..a3628cf2 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.24-openshift-4.22
+  tag: rhel-9-release-golang-1.25-openshift-4.22
diff --git a/Dockerfile.openshift b/Dockerfile.openshift
index c3160a6a..bde09650 100644
--- a/Dockerfile.openshift
+++ b/Dockerfile.openshift
@@ -1,5 +1,5 @@
 # This dockerfile is specific to building Multus admission controller for OpenShift
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.22 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
 
 # Add everything
 ENV PKG_NAME=github.com/k8snetworkplumbingwg/net-attach-def-admission-controller
