diff --git a/collection-scripts/gather b/collection-scripts/gather
index fea49aeb3..51445d85e 100755
--- a/collection-scripts/gather
+++ b/collection-scripts/gather
@@ -134,6 +134,10 @@ pids+=($!)
 /usr/bin/gather_nmstate &
 pids+=($!)
 
+# Gather Node Feature Discovery (NFD)
+/usr/bin/gather_nfd &
+pids+=($!)
+
 # Gather SR-IOV resources
 /usr/bin/gather_sriov &
 pids+=($!)
diff --git a/collection-scripts/gather_nfd b/collection-scripts/gather_nfd
new file mode 100755
index 000000000..06b734992
--- /dev/null
+++ b/collection-scripts/gather_nfd
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+source $(dirname "$0")/common.sh
+BASE_COLLECTION_PATH="must-gather"
+get_operator_ns "nfd"
+get_log_collection_args
+
+function get_nfd_crs() {
+    declare -a NFD_CRDS=("nodefeaturediscoveries" "nodefeaturegroups" "nodefeaturerules" "nodefeatures")
+
+    for CRD in "${NFD_CRDS[@]}"; do
+        echo "INFO: Collecting NFD Data for ${CRD} in namespace ${operator_ns}"
+        oc adm inspect ${log_collection_args} --dest-dir "${BASE_COLLECTION_PATH}" "${CRD}" -n "${operator_ns}"
+    done
+}
+
+oc adm inspect ${log_collection_args} --dest-dir "${BASE_COLLECTION_PATH}" "ns/${operator_ns}"
+get_nfd_crs
+
+# force disk flush to ensure that all data gathered is accessible in the copy container
+sync
