diff --git a/collection-scripts/gather_etcd b/collection-scripts/gather_etcd
index c89d7c1c..5846d54d 100755
--- a/collection-scripts/gather_etcd
+++ b/collection-scripts/gather_etcd
@@ -15,8 +15,12 @@ RUNNING_ETCD_POD=$(oc get pod -n openshift-etcd --no-headers -l app=etcd | grep
 # We cannot rely on ETCDCTL_ENDPOINTS on container because it may contain bootstrap VM
 ETCDCTL_ENDPOINTS=$(oc exec "${RUNNING_ETCD_POD}" -n openshift-etcd -c ${ETCDCTL_CONTAINER} -- /bin/sh -c "etcdctl member list" | awk -F', ' '{printf "%s%s",sep,$5; sep=","}')
 
+function ocp4etcd {
+	oc -n openshift-etcd exec -c "${ETCDCTL_CONTAINER}" "${RUNNING_ETCD_POD}" -- env "ETCDCTL_ENDPOINTS=${ETCDCTL_ENDPOINTS}" sh -c "$*"
+}
+
 function ocp4etcdctl {
-	oc -n openshift-etcd exec -c "${ETCDCTL_CONTAINER}" "${RUNNING_ETCD_POD}" -- env "ETCDCTL_ENDPOINTS=${ETCDCTL_ENDPOINTS}" etcdctl "$@"
+	ocp4etcd etcdctl "$*"
 }
 
 if [[ -z $RUNNING_ETCD_POD ]]; then
@@ -56,9 +60,9 @@ echo "INFO: Getting etcdctl alarm list"
 ocp4etcdctl alarm list -w json >${ETCD_LOG_PATH}/alarm_list.json &
 PIDS+=($!)
 
-# object count
-echo "INFO: Getting object count"
-ocp4etcdctl get / --prefix --keys-only | sed '/^$/d' | cut -d/ -f3 | sort | uniq -c | sort -rn | awk '{print $2 ":" $1}' | jq -nRc ' [inputs | index(":") as $ix | {(.[:$ix]): .[$ix+1:]}] | add' >${ETCD_LOG_PATH}/object_count.json &
+# object count and size
+echo "INFO: Getting object count and size"
+ocp4etcd "etcdctl get / --prefix --keys-only | grep -oE \"^/[a-z|.]+/[a-z|.|8]*\" | sort | uniq -c | sort -rn | while read KEY; do printf \"\$KEY\\t\" && etcdctl get \${KEY##* } --prefix --print-value-only | wc -c | numfmt --to=iec ; done | sort -k3 -hr | column -J -n \"etcd_resources\" -N \"object_count,object_name,object_size\"" | jq -c > ${ETCD_LOG_PATH}/object_count.json &
 PIDS+=($!)
 
 echo "INFO: Waiting for etcd info collection to complete ..."
