diff --git a/pkg/cmd/provisioning/aws/create_identity_provider.go b/pkg/cmd/provisioning/aws/create_identity_provider.go
index f49c4085b9..b00722c5ef 100644
--- a/pkg/cmd/provisioning/aws/create_identity_provider.go
+++ b/pkg/cmd/provisioning/aws/create_identity_provider.go
@@ -457,6 +457,9 @@ func getPartition(region string) (string, error) {
 	if strings.HasPrefix(region, "cn-") {
 		return "aws-cn", nil
 	}
+	if strings.HasPrefix(region, "eusc-") {
+		return "aws-eusc", nil
+	}
 	return "aws", nil
 }
 
