diff --git a/assets/monitoring-plugin/deployment.yaml b/assets/monitoring-plugin/deployment.yaml
index cad74b604f..aa24effe5c 100644
--- a/assets/monitoring-plugin/deployment.yaml
+++ b/assets/monitoring-plugin/deployment.yaml
@@ -46,7 +46,7 @@ spec:
       automountServiceAccountToken: true
       containers:
       - args:
-        - --config-path=/opt/app-root/web/dist
+        - --config-path=/opt/app-root/config
         - --static-path=/opt/app-root/web/dist
         - --cert=/var/cert/tls.crt
         - --key=/var/cert/tls.key
diff --git a/jsonnet/components/monitoring-plugin.libsonnet b/jsonnet/components/monitoring-plugin.libsonnet
index c2122eb2c3..a9a49ef69d 100644
--- a/jsonnet/components/monitoring-plugin.libsonnet
+++ b/jsonnet/components/monitoring-plugin.libsonnet
@@ -197,7 +197,7 @@ function(params)
                   $.volumeMount(tlsVolumeName, tlsMountPath),
                 ],
                 args: [
-                  '--config-path=/opt/app-root/web/dist',
+                  '--config-path=/opt/app-root/config',
                   '--static-path=/opt/app-root/web/dist',
                   '--cert=' + tlsCertPath,
                   '--key=' + tlsKeyPath,
