diff --git a/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx b/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
index 307c040e05..de4a922d44 100644
--- a/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterWizard/ClusterDetailsForm.tsx
@@ -165,11 +165,11 @@ const ClusterDetailsForm = (props: ClusterDetailsFormProps) => {
               {!isSingleClusterFeatureEnabled && (
                 <GridItem>
                   <Flex alignItems={{ default: 'alignItemsCenter' }} gap={{ default: 'gapSm' }}>
-                    <TechnologyPreview />
                     <InstallDisconnectedSwitch isDisabled={!!cluster} />
                     <span>
                       {t("ai:I'm installing on a disconnected/air-gapped/secured environment")}
                     </span>
+                    <TechnologyPreview />
                   </Flex>
                 </GridItem>
               )}
diff --git a/libs/ui-lib/lib/ocm/components/clusterWizard/disconnected/BasicStep.tsx b/libs/ui-lib/lib/ocm/components/clusterWizard/disconnected/BasicStep.tsx
index 7ac86f3cad..8e2cd00954 100644
--- a/libs/ui-lib/lib/ocm/components/clusterWizard/disconnected/BasicStep.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterWizard/disconnected/BasicStep.tsx
@@ -116,9 +116,9 @@ const BasicStepForm = () => {
           </GridItem>
           <GridItem>
             <Flex alignItems={{ default: 'alignItemsCenter' }} gap={{ default: 'gapSm' }}>
-              <TechnologyPreview />
               <InstallDisconnectedSwitch isDisabled={!!clusterId} />
               <span>{t("ai:I'm installing on a disconnected/air-gapped/secured environment")}</span>
+              <TechnologyPreview />
             </Flex>
           </GridItem>
           <GridItem>
