diff --git a/OWNERS b/OWNERS
index 07cbf8f41..2194b952e 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,16 +1,11 @@
 reviewers:
-  - bparees
-  - deads2k
-  - soltysh
+  - control-plane-approvers
   - divyansh42
   - sayan-biswas
   - prabhapa
   - moebasim
 approvers:
-  - bparees
-  - deads2k
-  - soltysh
-  - ingvagabund
+  - control-plane-approvers
   - sayan-biswas
   - adambkaplan
   - prabhapa
diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
new file mode 100644
index 000000000..66464a003
--- /dev/null
+++ b/OWNERS_ALIASES
@@ -0,0 +1,16 @@
+aliases:
+  control-plane-approvers:
+    - ardaguclu
+    - atiratree
+    - benluddy
+    - bertinatto
+    - everettraven
+    - flavianmissi
+    - gangwgr
+    - ingvagabund
+    - kaleemsiddiqu
+    - p0lyn0mial
+    - rh-roman
+    - ricardomaraschini
+    - tjungblu
+    - xueqzhan
