diff --git a/Dockerfile.ocp b/Dockerfile.ocp
index 2a73dfdea..4fa2d4cc3 100644
--- a/Dockerfile.ocp
+++ b/Dockerfile.ocp
@@ -51,6 +51,9 @@ RUN --mount=from=wheel-builder,source=/wheels,target=/wheels \
     /bin/prepare-ipxe.sh && \
     rm -f /bin/prepare-ipxe.sh
 
+# Add PQC support until we have pqc enabled ubi minimal images available
+RUN update-crypto-policies --set DEFAULT:PQ
+
 # IRONIC #
 COPY --from=builder /tmp/uefi_esp_*.img /templates/
 
diff --git a/Dockerfile.scos b/Dockerfile.scos
index 0f8e486ed..31561bd22 100644
--- a/Dockerfile.scos
+++ b/Dockerfile.scos
@@ -26,6 +26,9 @@ RUN prepare-image.sh && \
     /bin/prepare-ipxe.sh && \
     rm -f /bin/prepare-ipxe.sh
 
+# Add PQC support until we have pqc enabled ubi minimal images available
+RUN update-crypto-policies --set DEFAULT:PQ
+
 # IRONIC #
 COPY --from=builder /tmp/uefi_esp_*.img /tmp/
 
diff --git a/main-packages-list.ocp b/main-packages-list.ocp
index 595583df8..94ff2a0fa 100644
--- a/main-packages-list.ocp
+++ b/main-packages-list.ocp
@@ -88,3 +88,4 @@ qemu-img
 sqlite
 util-linux
 xorriso
+crypto-policies-scripts
\ No newline at end of file
diff --git a/main-packages-list.okd b/main-packages-list.okd
index 45cbdf27e..81843be3f 100644
--- a/main-packages-list.okd
+++ b/main-packages-list.okd
@@ -15,3 +15,4 @@ qemu-img
 sqlite
 util-linux
 xorriso
+crypto-policies-scripts
\ No newline at end of file
