diff --git a/Dockerfile.control-plane b/Dockerfile.control-plane
index bf82499bc9a6..937bf2d6b26a 100644
--- a/Dockerfile.control-plane
+++ b/Dockerfile.control-plane
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.23 AS builder
+FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.25-openshift-4.23 AS builder
 ARG COMMIT_HASH
 
 WORKDIR /hypershift
