diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index a3628cf24..1d88a59fd 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.25-openshift-4.22
+  tag: rhel-9-release-golang-1.26-openshift-5.0
diff --git a/Dockerfile.rhel b/Dockerfile.rhel
index 04f3202ec..f8fbdd7b7 100644
--- a/Dockerfile.rhel
+++ b/Dockerfile.rhel
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 
 WORKDIR /workspace
 COPY . .
@@ -7,7 +7,7 @@ RUN CGO_ENABLED=0 GOOS=${GOOS} GOPROXY=${GOPROXY} go build \
   -o manager \
   cmd/manager/main.go
 
-FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
+FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
 
 LABEL description="OpenStack Resource Controller Manager"
 COPY --from=builder /workspace/manager .
