diff --git a/pkg/manifests/manifests.go b/pkg/manifests/manifests.go
index 276b8a7020..af396d707e 100644
--- a/pkg/manifests/manifests.go
+++ b/pkg/manifests/manifests.go
@@ -2693,16 +2693,6 @@ func (f *Factory) NewConfigMap(manifest []byte) (*v1.ConfigMap, error) {
 	return &cm, nil
 }
 
-func (f *Factory) NewConfigMapList(manifest []byte) (*v1.ConfigMapList, error) {
-	cml := v1.ConfigMapList{}
-	err := decodeYAML(manifest, &cml)
-	if err != nil {
-		return nil, err
-	}
-
-	return &cml, nil
-}
-
 func (f *Factory) NewServiceAccount(manifest []byte) (*v1.ServiceAccount, error) {
 	sa := v1.ServiceAccount{}
 	err := decodeYAML(manifest, &sa)
