diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index 7987ec053..f28d4298a 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -1,23 +1,14 @@
 aliases:
   multiarch-approvers:
-  - jaypoulz
   - Prashanth684
   - tvardema
-  - aleskandro
-  - jeffdyoung
-  - andymcc
-  - SaravanaStorageNetwork
-  - r4f4
+  - prb112
+  - mjturek
+  - hamzy
   multiarch-reviewers:
-  - jaypoulz
   - Prashanth684
   - tvardema
-  - aleskandro
-  - jeffdyoung
-  - andymcc
-  - SaravanaStorageNetwork
-  - r4f4
-  - alkleinrh
   - manojnkumar
+  - prb112
+  - mjturek
   - hamzy
-  - Shilpa-Gokul
