diff --git a/test/extended-priv/mco_drain.go b/test/extended-priv/mco_drain.go
index 2e4adca596..7a71686d67 100644
--- a/test/extended-priv/mco_drain.go
+++ b/test/extended-priv/mco_drain.go
@@ -281,8 +281,8 @@ var _ = g.Describe("[sig-mco][Suite:openshift/machine-config-operator/longdurati
 
 		exutil.By("Set maxUnavailable value")
 		maxUnavailable := 2
+		defer mcp.SetSpec(mcp.GetSpecOrFail())
 		mcp.SetMaxUnavailable(maxUnavailable)
-		defer mcp.RemoveMaxUnavailable()
 
 		exutil.By("Create a MC to deploy a config file")
 		filePath := "/etc/TC-49672-mco-test-file-order"
diff --git a/test/extended-priv/mco_machineconfigpool.go b/test/extended-priv/mco_machineconfigpool.go
index 2a67c6fd34..85dd389e6d 100644
--- a/test/extended-priv/mco_machineconfigpool.go
+++ b/test/extended-priv/mco_machineconfigpool.go
@@ -339,8 +339,8 @@ var _ = g.Describe("[sig-mco][Suite:openshift/machine-config-operator/longdurati
 		}
 
 		exutil.By("Set the maxUnavailable value to 2")
+		defer mcp.SetSpec(mcp.GetSpecOrFail())
 		mcp.SetMaxUnavailable(2)
-		defer mcp.RemoveMaxUnavailable()
 		logger.Infof("OK!\n")
 
 		exutil.By("Manually cordon one of the nodes")
