diff --git a/cmd/machine-api-tests-ext/main.go b/cmd/machine-api-tests-ext/main.go
index 7f5b625b6f..5f7eec5ba6 100644
--- a/cmd/machine-api-tests-ext/main.go
+++ b/cmd/machine-api-tests-ext/main.go
@@ -15,6 +15,7 @@ import (
 	"k8s.io/component-base/logs"
 
 	// If using ginkgo, import your tests here
+	_ "github.com/openshift/machine-api-operator/test/e2e/nutanix"
 	_ "github.com/openshift/machine-api-operator/test/e2e/vsphere"
 )
 
diff --git a/test/e2e/nutanix/multi-subnet.go b/test/e2e/nutanix/multi-subnet.go
index abd665d09f..dcbe774a41 100644
--- a/test/e2e/nutanix/multi-subnet.go
+++ b/test/e2e/nutanix/multi-subnet.go
@@ -295,7 +295,7 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:NutanixMultiSubnets][pl
 		Expect(err).NotTo(HaveOccurred())
 		Expect(len(infra.Spec.PlatformSpec.Nutanix.FailureDomains)).ShouldNot(Equal(0))
 
-		machineNetworks = append(machineNetworks, infra.Spec.PlatformSpec.Nutanix.FailureDomains...)
+		machineNetworks = infra.Spec.PlatformSpec.Nutanix.FailureDomains
 		Expect(len(machineNetworks)).ShouldNot(Equal(0))
 
 		nodes, err = c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
