diff --git a/libs/ui-lib/lib/ocm/components/clusters/ClusterBreadcrumbs.tsx b/libs/ui-lib/lib/ocm/components/clusters/ClusterBreadcrumbs.tsx
new file mode 100644
index 0000000000..42f3a815c0
--- /dev/null
+++ b/libs/ui-lib/lib/ocm/components/clusters/ClusterBreadcrumbs.tsx
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Breadcrumb, BreadcrumbItem, PageSection } from '@patternfly/react-core';
+import { Link } from 'react-router-dom-v5-compat';
+import { isInOcm } from '../../../common/api';
+
+const ClusterBreadcrumbs = ({ clusterName }: { clusterName?: string }) => {
+  return (
+    !isInOcm && (
+      <PageSection hasBodyWrapper={false}>
+        <Breadcrumb data-testid="cluster-breadcrumbs">
+          <BreadcrumbItem data-testid="assisted-clusters-breadcrumb">
+            <Link to={`..`}>Assisted Clusters</Link>
+          </BreadcrumbItem>
+          {clusterName && (
+            <BreadcrumbItem data-testid="cluster-breadcrumb" isActive>
+              {clusterName}
+            </BreadcrumbItem>
+          )}
+        </Breadcrumb>
+      </PageSection>
+    )
+  );
+};
+
+export default ClusterBreadcrumbs;
diff --git a/libs/ui-lib/lib/ocm/components/clusters/ClusterPage.tsx b/libs/ui-lib/lib/ocm/components/clusters/ClusterPage.tsx
index 80f44166c3..d9fc925412 100644
--- a/libs/ui-lib/lib/ocm/components/clusters/ClusterPage.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusters/ClusterPage.tsx
@@ -14,6 +14,7 @@ import CancelInstallationModal from '../clusterDetail/CancelInstallationModal';
 import ResetClusterModal from '../clusterDetail/ResetClusterModal';
 import { AddHosts } from '../AddHosts';
 import { ClusterDefaultConfigurationProvider } from '../clusterConfiguration/ClusterDefaultConfigurationContext';
+import ClusterBreadcrumbs from './ClusterBreadcrumbs';
 import ClusterWizard from '../clusterWizard/ClusterWizard';
 import { ModalDialogsContextProvider } from '../hosts/ModalDialogsContext';
 import { useClusterPolling, useFetchCluster } from './clusterPolling';
@@ -36,11 +37,11 @@ import { OpenShiftVersionsContextProvider } from '../clusterWizard/OpenShiftVers
 
 const ClusterPageGeneric = ({
   clusterId,
-  showHeader = false,
+  showBreadcrumbs = false,
   resetModal,
 }: {
   clusterId: string;
-  showHeader?: boolean;
+  showBreadcrumbs?: boolean;
   resetModal?: React.ReactNode;
 }) => {
   const fetchCluster = useFetchCluster(clusterId);
@@ -88,6 +89,7 @@ const ClusterPageGeneric = ({
     ) {
       return (
         <>
+          {showBreadcrumbs && <ClusterBreadcrumbs clusterName={cluster.name} />}
           <PageSection hasBodyWrapper={false}>
             <Content component="h1">{cluster.name}</Content>
           </PageSection>
@@ -99,7 +101,8 @@ const ClusterPageGeneric = ({
     } else {
       return (
         <>
-          {showHeader && (
+          {showBreadcrumbs && <ClusterBreadcrumbs clusterName={cluster.name} />}
+          {showBreadcrumbs && (
             <PageSection hasBodyWrapper={false}>
               <AssistedInstallerHeader />
             </PageSection>
@@ -211,7 +214,7 @@ export const ClusterPage = () => {
   const { clusterId } = useParams<{ clusterId: string }>();
   return (
     <AlertsContextProvider>
-      <ClusterPageGeneric clusterId={clusterId || ''} showHeader />
+      <ClusterPageGeneric clusterId={clusterId || ''} showBreadcrumbs />
     </AlertsContextProvider>
   );
 };
diff --git a/libs/ui-lib/lib/ocm/components/clusters/NewClusterPage.tsx b/libs/ui-lib/lib/ocm/components/clusters/NewClusterPage.tsx
index 0b7d3b2eeb..8c41913816 100644
--- a/libs/ui-lib/lib/ocm/components/clusters/NewClusterPage.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusters/NewClusterPage.tsx
@@ -1,6 +1,7 @@
 import React from 'react';
 import { PageSection } from '@patternfly/react-core';
 import { AlertsContextProvider } from '../../../common';
+import ClusterBreadcrumbs from './ClusterBreadcrumbs';
 import { ClusterDefaultConfigurationProvider } from '../clusterConfiguration/ClusterDefaultConfigurationContext';
 import NewClusterWizard from '../clusterWizard/NewClusterWizard';
 import ClusterWizardContextProvider from '../clusterWizard/ClusterWizardContextProvider';
@@ -12,7 +13,7 @@ import { AssistedInstallerHeader } from './AssistedInstallerHeader';
 import { ModalDialogsContextProvider } from '../hosts/ModalDialogsContext';
 import { OpenShiftVersionsContextProvider } from '../clusterWizard/OpenShiftVersionsContext';
 
-export const NewClusterPage = ({ children }: React.PropsWithChildren<unknown>) => {
+export const NewClusterPage = () => {
   return (
     <AlertsContextProvider>
       <SentryErrorMonitorContextProvider>
@@ -23,7 +24,7 @@ export const NewClusterPage = ({ children }: React.PropsWithChildren<unknown>) =
           >
             <OpenShiftVersionsContextProvider>
               <NewFeatureSupportLevelProvider loadingUi={<ClusterLoading />}>
-                {children}
+                <ClusterBreadcrumbs clusterName="New cluster" />
                 <PageSection hasBodyWrapper={false} isFilled>
                   <AssistedInstallerHeader />
                 </PageSection>
