diff --git a/pkg/testsuites/standard_suites.go b/pkg/testsuites/standard_suites.go
index 3028c929e32a..6c3c2258cb55 100644
--- a/pkg/testsuites/standard_suites.go
+++ b/pkg/testsuites/standard_suites.go
@@ -191,7 +191,7 @@ var staticSuites = []ginkgo.TestSuite{
 		Tests that exercise the OpenShift image-registry functionality.
 		`),
 		Qualifiers: []string{
-			withStandardEarlyOrLateTests("name.contains('[sig-imageregistry]') && !name.contains('[Local]')"),
+			withStandardEarlyOrLateTests("name.contains('[sig-imageregistry]') && !name.contains('[Local]') && !name.contains('[Disruptive]') && !name.contains('[Serial]')"),
 		},
 	},
 	{
