diff --git a/OWNERS b/OWNERS
index a0baecfa7..ae297e24a 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,9 +1,5 @@
 reviewers:
-  - atiratree
-  - deads2k
-  - ingvagabund
+  - control-plane-approvers
 approvers:
-  - atiratree
-  - deads2k
-  - ingvagabund
+  - control-plane-approvers
 component: "kube-controller-manager"
diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
new file mode 100644
index 000000000..66464a003
--- /dev/null
+++ b/OWNERS_ALIASES
@@ -0,0 +1,16 @@
+aliases:
+  control-plane-approvers:
+    - ardaguclu
+    - atiratree
+    - benluddy
+    - bertinatto
+    - everettraven
+    - flavianmissi
+    - gangwgr
+    - ingvagabund
+    - kaleemsiddiqu
+    - p0lyn0mial
+    - rh-roman
+    - ricardomaraschini
+    - tjungblu
+    - xueqzhan
