diff --git a/Makefile b/Makefile
index 2bd69417b83..dac061a16dd 100644
--- a/Makefile
+++ b/Makefile
@@ -130,8 +130,16 @@ verify-git-clean:
 	$(eval STATUS = $(shell git status -s))
 	$(if $(strip $(STATUS)),$(error untracked files detected: ${STATUS}))
 
+.PHONY: verify-codecov
+verify-codecov: ## Validate codecov.yml against Codecov's API.
+	@curl --silent --show-error \
+		--connect-timeout 5 --max-time 30 \
+		--retry 3 --retry-all-errors --retry-delay 1 \
+		--data-binary @codecov.yml https://codecov.io/validate \
+		| tee /dev/stderr | grep -q "^Valid!"
+
 .PHONY: verify-parallel
-verify-parallel: verify-codespell lint cpo-container-sync run-gitlint
+verify-parallel: verify-codespell verify-codecov lint cpo-container-sync run-gitlint
 
 .PHONY: verify
 verify: generate update staticcheck fmt vet
