diff --git a/pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go b/pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go
index 3908ff1817cc..1320bcc74e00 100644
--- a/pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go
+++ b/pkg/monitortests/clusterversionoperator/terminationmessagepolicy/monitortest.go
@@ -139,7 +139,6 @@ func (w *terminationMessagePolicyChecker) CollectData(ctx context.Context, stora
 		// they should be fixed and removed from here:
 		"openshift-backplane":                sets.NewString("pods/osd-delete-backplane-serviceaccounts"), // filed OCPBUGS-84527 to fix
 		"openshift-cloud-controller-manager": sets.NewString("pods/aws-cloud-controller-manager"),         // filed OCPBUGS-84512 to fix
-		"openshift-cluster-machine-approver": sets.NewString("pods/machine-approver-capi"),                // filed OCPBUGS-84521 to fix
 		"openshift-cluster-version":          sets.NewString("pods/version--"),                            // filed OCPBUGS-84513 to fix
 		"openshift-cnv": sets.NewString( // filed OCPBUGS-84522 to fix
 			"pods/hostpath-provisioner-operator",
