diff --git a/pkg/test/extensions/binary.go b/pkg/test/extensions/binary.go
index 94eba6fe2ec5..7f322a4a174d 100644
--- a/pkg/test/extensions/binary.go
+++ b/pkg/test/extensions/binary.go
@@ -333,6 +333,10 @@ var extensionBinaries = []TestBinary{
 		imageTag:   "service-ca-operator",
 		binaryPath: "/usr/bin/service-ca-operator-tests-ext.gz",
 	},
+	{
+		imageTag:   "vsphere-csi-driver-operator",
+		binaryPath: "/usr/bin/vmware-vsphere-csi-driver-operator-tests-ext.gz",
+	},
 }
 
 // Info returns information about this particular extension.
