diff --git a/assets/volumedatasourcevalidator/04_deployment.yaml b/assets/volumedatasourcevalidator/04_deployment.yaml
index fbba39922..b218acd02 100644
--- a/assets/volumedatasourcevalidator/04_deployment.yaml
+++ b/assets/volumedatasourcevalidator/04_deployment.yaml
@@ -20,6 +20,14 @@ spec:
 
     spec:
       serviceAccountName: volume-data-source-validator
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - key: CriticalAddonsOnly
+        operator: Exists
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: "NoSchedule"
       containers:
         - name: volume-data-source-validator
           image: ${VOLUME_DATA_SOURCE_VALIDATOR_IMAGE}
diff --git a/pkg/operator/volumedatasourcevalidator/controller.go b/pkg/operator/volumedatasourcevalidator/controller.go
index f79047d13..724fb719f 100644
--- a/pkg/operator/volumedatasourcevalidator/controller.go
+++ b/pkg/operator/volumedatasourcevalidator/controller.go
@@ -113,9 +113,13 @@ func (c *VolumeDataSourceValidatorStarter) createVolumeDataSourceValidatorManage
 		clients.OperatorClient,
 		clients.KubeClient,
 		clients.KubeInformers.InformersFor(csoclients.OperatorNamespace).Apps().V1().Deployments(),
+	).WithExtraInformers(
+		clients.ConfigInformers.Config().V1().Infrastructures().Informer(),
 	).WithManifestHooks(
 		c.withReplacerHook(),
 		csidrivercontrollerservicecontroller.WithLeaderElectionReplacerHook(leConfig),
+	).WithDeploymentHooks(
+		csidrivercontrollerservicecontroller.WithControlPlaneTopologyHook(clients.ConfigInformers),
 	).WithConditions(
 		operatorapi.OperatorStatusTypeProgressing,
 		operatorapi.OperatorStatusTypeDegraded,
