diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index 284a9100..a3628cf2 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.24-openshift-4.22
+  tag: rhel-9-release-golang-1.25-openshift-4.22
diff --git a/Dockerfile.rhel b/Dockerfile.rhel
index e5107c1f..f6b5607d 100644
--- a/Dockerfile.rhel
+++ b/Dockerfile.rhel
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.22 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
 WORKDIR /go/src/github.com/openshift/machine-api-provider-powervs
 COPY . .
 # VERSION env gets set in the openshift/release image and refers to the golang version, which interfers with our own
