diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index 7d60bb1764..1d88a59fdf 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.26-openshift-4.23
+  tag: rhel-9-release-golang-1.26-openshift-5.0
diff --git a/operator-framework-tools.Dockerfile b/operator-framework-tools.Dockerfile
index 4b83461142..c7437a92f9 100644
--- a/operator-framework-tools.Dockerfile
+++ b/operator-framework-tools.Dockerfile
@@ -1,18 +1,18 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.26-openshift-4.23 AS builder-rhel8
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.26-openshift-5.0 AS builder-rhel8
 ENV GOPATH /go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
 WORKDIR /src
 COPY . .
 RUN make build/registry cross
 
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-4.23 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 ENV GOPATH /go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
 WORKDIR /src
 COPY . .
 RUN make build/registry cross
 
-FROM registry.ci.openshift.org/ocp/4.23:base-rhel9
+FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
 ENTRYPOINT ["sh", "-c", "echo 'Running this image is not supported' && exit 1"]
 # clear any default CMD
 CMD []
