diff --git a/OWNERS b/OWNERS
index 6c669ffd..0e7d9d09 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,8 +1,20 @@
 reviewers:
-  - sjenning
-  - enxebre
+  - bryan-cox
+  - clebs
   - csrwng
-approvers:
-  - sjenning
+  - devguyio
   - enxebre
+  - jparrill
+  - muraee
+  - Nirshal
+  - sdminonne
+  - sjenning
+approvers:
+  - bryan-cox
   - csrwng
+  - devguyio
+  - enxebre
+  - jparrill
+  - muraee
+  - sjenning
+component: hypershift
