diff --git a/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts b/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
index 5753531da2a..85ca2f12ff0 100644
--- a/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
+++ b/frontend/packages/integration-tests/tests/crud/customresourcedefinition.cy.ts
@@ -100,6 +100,7 @@ describe('CustomResourceDefinitions', () => {
     listPage.dvRows.shouldBeLoaded();
     listPage.dvRows.clickKebabAction(`CRD${testName}`, 'View instances');
     cy.url().should('include', `/k8s/all-namespaces/test.example.com~v1~CRD${testName}`);
+    cy.byTestID('empty-box').should('be.visible');
     listPage.isCreateButtonVisible();
     listPage.clickCreateYAMLbutton();
     yamlEditor.isLoaded();
