diff --git a/web/src/components/query-browser.tsx b/web/src/components/query-browser.tsx
index 77bae35b8..74924df9b 100644
--- a/web/src/components/query-browser.tsx
+++ b/web/src/components/query-browser.tsx
@@ -124,7 +124,6 @@ const GraphEmptyState: FC<GraphEmptyStateProps> = ({ children, title }) => (
   </div>
 );
 
-// eslint-disable-next-line react/display-name
 const SpanControls: FC<SpanControlsProps> = memo(
   ({ defaultSpanText, onChange, span, hasReducedResolution }) => {
     const { t } = useTranslation(process.env.I18N_NAMESPACE);
@@ -218,6 +217,8 @@ const SpanControls: FC<SpanControlsProps> = memo(
   },
 );
 
+SpanControls.displayName = 'SpanControls';
+
 const LegendContainer = ({ children }: { children?: ReactNode }) => {
   // The first child should be a <rect> with a `width` prop giving the legend's content width
   const width = children?.[0]?.props?.width ?? '100%';
@@ -240,7 +241,6 @@ const getXDomain = (endTime: number, span: number): AxisDomain => [endTime - spa
 
 const ONE_MINUTE = 60 * 1000;
 
-// eslint-disable-next-line react/display-name
 const Graph: FC<GraphProps> = memo(
   ({
     allSeries,
@@ -426,6 +426,8 @@ const Graph: FC<GraphProps> = memo(
   },
 );
 
+Graph.displayName = 'Graph';
+
 const formatSeriesValues = (
   values: PrometheusValue[],
   samples: number,
diff --git a/web/src/components/targets-page.tsx b/web/src/components/targets-page.tsx
index 187fe3cb7..8bf8e2d10 100644
--- a/web/src/components/targets-page.tsx
+++ b/web/src/components/targets-page.tsx
@@ -212,7 +212,7 @@ const ServiceMonitor: FC<{ target: Target }> = ({ target }) => {
   );
 };
 
-// eslint-disable-next-line react/display-name, react/prop-types
+// eslint-disable-next-line react/prop-types
 const Health: FC<{ health: 'up' | 'down' }> = memo(({ health }) => {
   const { t } = useTranslation(process.env.I18N_NAMESPACE);
 
@@ -227,6 +227,8 @@ const Health: FC<{ health: 'up' | 'down' }> = memo(({ health }) => {
   );
 });
 
+Health.displayName = 'Health';
+
 type WatchErrorAlertProps = {
   loadError: { code: number; message: string };
   title: string;
