diff --git a/OWNERS_ALIASES b/OWNERS_ALIASES
index d73ad570322..d9467f9db7f 100644
--- a/OWNERS_ALIASES
+++ b/OWNERS_ALIASES
@@ -15,6 +15,7 @@ aliases:
     - sjenning
     - muraee
     - bryan-cox
+    - cblecker
     - jparrill
     - devguyio
   core-reviewers:
@@ -23,6 +24,7 @@ aliases:
     - sjenning
     - muraee
     - bryan-cox
+    - cblecker
     - jparrill
     - devguyio
   karpenter-approvers:
