diff --git a/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx b/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
index e8a87f5a15..23f4c64b6f 100644
--- a/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
@@ -146,7 +146,7 @@ export const CustomManifestsPage = ({
           </span>
         </Tooltip>
       </FormGroup>
-      {!useCustomManifests && <div style={{ minHeight: '120px' }} aria-hidden="true" />}
+      {!useCustomManifests && <div style={{ minHeight: '250px' }} aria-hidden="true" />}
       {useCustomManifests && (
         <>
           {reminderAlerts}
