diff --git a/test/e2e/util/util.go b/test/e2e/util/util.go
index bf8afe1dc9c..da922886925 100644
--- a/test/e2e/util/util.go
+++ b/test/e2e/util/util.go
@@ -1614,6 +1614,7 @@ func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx conte
 			"openstack-manila-csi":                   "app",
 			"karpenter":                              "app",
 			"karpenter-operator":                     "app",
+			"aws-node-termination-handler":           "app",
 		}
 
 		hcpPods := &corev1.PodList{}
