diff --git a/lib/resourcebuilder/apps.go b/lib/resourcebuilder/apps.go
index 6d05f41a1..af1e252ab 100644
--- a/lib/resourcebuilder/apps.go
+++ b/lib/resourcebuilder/apps.go
@@ -46,7 +46,7 @@ func (b *builder) modifyDeployment(ctx context.Context, deployment *appsv1.Deplo
 		if err != nil && !errors.IsNotFound(err) {
 			return err
 		}
-		if !errors.IsNotFound(err) {
+		if err == nil {
 			lbURL, err := url.Parse(infrastructureConfig.Status.APIServerInternalURL)
 			if err != nil {
 				return err
