diff --git a/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx b/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
index ee2314410f..56ddb2bcd2 100644
--- a/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/CustomManifestsPage.tsx
@@ -78,7 +78,10 @@ export const CustomManifestsPage = ({
     await ClustersService.removeClusterManifests(manifests, cluster.id);
     setUseCustomManifests(false);
     setDeleteModalOpen(false);
-    await clusterWizardContext.updateUISettings({ customManifestsAdded: false });
+    await clusterWizardContext.updateUISettings({
+      customManifestsAdded: false,
+      customManifestsUpdated: true,
+    });
   }, [cluster.id, clusterWizardContext]);
 
   const fieldId = getFieldId('use-custom-manifests', 'switch');
diff --git a/libs/ui-lib/lib/ocm/components/clusterConfiguration/review/ReviewStep.tsx b/libs/ui-lib/lib/ocm/components/clusterConfiguration/review/ReviewStep.tsx
index f27349ede8..1ec318d3ba 100644
--- a/libs/ui-lib/lib/ocm/components/clusterConfiguration/review/ReviewStep.tsx
+++ b/libs/ui-lib/lib/ocm/components/clusterConfiguration/review/ReviewStep.tsx
@@ -24,19 +24,17 @@ const ReviewStep = ({ cluster }: { cluster: Cluster }) => {
   const [disableInstallButton, setDisableInstallButton] = React.useState<boolean>();
 
   React.useEffect(() => {
-    if (
+    const manifestFail =
       cluster['last-installation-preparation']?.status === 'failed' &&
-      cluster['last-installation-preparation'].reason?.includes('manifest')
-    ) {
-      if (clusterWizardContext.uiSettings?.customManifestsUpdated === undefined) {
-        setDisableInstallButton(true);
-        void clusterWizardContext.updateUISettings({ customManifestsUpdated: false });
-      } else if (clusterWizardContext.uiSettings?.customManifestsUpdated === false) {
-        setDisableInstallButton(true);
-      }
-    } else {
-      setDisableInstallButton(isViewerMode || isStartingInstallation || cluster.status !== 'ready');
-    }
+      cluster['last-installation-preparation'].reason?.includes('manifest');
+    const userDidntAddressManifestIssue =
+      clusterWizardContext.uiSettings?.customManifestsUpdated !== true;
+    setDisableInstallButton(
+      (manifestFail && userDidntAddressManifestIssue) ||
+        isViewerMode ||
+        isStartingInstallation ||
+        cluster.status !== 'ready',
+    );
   }, [cluster, isStartingInstallation, isViewerMode, clusterWizardContext]);
 
   const handleClusterInstall = async () => {
