diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index e307e5af66..a3628cf240 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.24-openshift-4.21
+  tag: rhel-9-release-golang-1.25-openshift-4.22
diff --git a/openshift/Dockerfile b/openshift/Dockerfile
index 12a00e4c70..c53956a08a 100644
--- a/openshift/Dockerfile
+++ b/openshift/Dockerfile
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.21 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
 
 WORKDIR /build
 # NOTE: We're copying v2 in like this rather than using v2 as the context
@@ -9,7 +9,7 @@ RUN GO111MODULE=on CGO_ENABLED=0 GOOS=${GOOS} GOPROXY=${GOPROXY} go build \
   -o=azure-service-operator \
   ./cmd/controller
 
-FROM registry.ci.openshift.org/ocp/4.21:base-rhel9
+FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
 
 LABEL description="Azure Service Operator Controller Manager"
 
