diff --git a/Dockerfile.github-actions-runner b/Dockerfile.github-actions-runner
index 4d3f73b81482..aedf49d13870 100644
--- a/Dockerfile.github-actions-runner
+++ b/Dockerfile.github-actions-runner
@@ -19,7 +19,7 @@ RUN apt-get update && \
     && rm -rf /var/lib/apt/lists/*
 
 ARG TARGETARCH
-ARG GO_VERSION=1.25.7
+ARG GO_VERSION=1.26.4
 RUN curl -fsSL "https://go.dev/dl/go${GO_VERSION}.linux-${TARGETARCH}.tar.gz" | tar -C /usr/local -xz
 
 RUN OCP_ARCH=$([ "$TARGETARCH" = "amd64" ] && echo "x86_64" || echo "aarch64") && \
