diff --git a/manifests/machineconfigcontroller/deployment.yaml b/manifests/machineconfigcontroller/deployment.yaml
index 002b1881e6..0049d33daa 100644
--- a/manifests/machineconfigcontroller/deployment.yaml
+++ b/manifests/machineconfigcontroller/deployment.yaml
@@ -31,6 +31,21 @@ spec:
             cpu: 20m
             memory: 50Mi
         terminationMessagePolicy: FallbackToLogsOnError
+        {{if .ControllerConfig.Proxy}}
+        env:
+        {{if .ControllerConfig.Proxy.HTTPProxy}}
+        - name: HTTP_PROXY
+          value: {{.ControllerConfig.Proxy.HTTPProxy}}
+        {{end}}
+        {{if .ControllerConfig.Proxy.HTTPSProxy}}
+        - name: HTTPS_PROXY
+          value: {{.ControllerConfig.Proxy.HTTPSProxy}}
+        {{end}}
+        {{if .ControllerConfig.Proxy.NoProxy}}
+        - name: NO_PROXY
+          value: "{{.ControllerConfig.Proxy.NoProxy}}"
+        {{end}}
+        {{end}}
       - name: kube-rbac-proxy
         image: {{.Images.KubeRbacProxy}}
         ports:
diff --git a/pkg/operator/render_test.go b/pkg/operator/render_test.go
index 2cbaff6cb3..4602089634 100644
--- a/pkg/operator/render_test.go
+++ b/pkg/operator/render_test.go
@@ -279,6 +279,30 @@ func TestRenderAsset(t *testing.T) {
 				"--payload-version=4.8.0-rc.0",
 			},
 		},
+		{
+			// Test that MCC deployment is rendered correctly with proxy config
+			Path: "manifests/machineconfigcontroller/deployment.yaml",
+			RenderConfig: &renderConfig{
+				TargetNamespace: "testing-namespace",
+				ReleaseVersion:  "4.8.0-rc.0",
+				Images: &ctrlcommon.RenderConfigImages{
+					MachineConfigOperator: "mco-operator-image",
+					KubeRbacProxy:         "kube-rbac-proxy-image",
+				},
+				ControllerConfig: mcfgv1.ControllerConfigSpec{
+					Proxy: &configv1.ProxyStatus{
+						HTTPSProxy: "https://i.am.a.proxy.server",
+						NoProxy:    "*", // See: https://bugzilla.redhat.com/show_bug.cgi?id=1947066
+					},
+				},
+			},
+			FindExpected: []string{
+				"image: mco-operator-image",
+				"- name: HTTPS_PROXY\n          value: https://i.am.a.proxy.server",
+				"- name: NO_PROXY\n          value: \"*\"", // Ensure the * is quoted: "*": https://bugzilla.redhat.com/show_bug.cgi?id=1947066
+				"--payload-version=4.8.0-rc.0",
+			},
+		},
 		{
 			// Bad path, will cause asset error
 			Path:  "BAD PATH",
