diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index a3628cf24..1d88a59fd 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.25-openshift-4.22
+  tag: rhel-9-release-golang-1.26-openshift-5.0
diff --git a/Dockerfile.rhel7 b/Dockerfile.rhel7
index 425dc5048..99bf1681e 100644
--- a/Dockerfile.rhel7
+++ b/Dockerfile.rhel7
@@ -1,11 +1,11 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.25-openshift-4.22 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 WORKDIR /go/src/github.com/openshift/service-ca-operator
 COPY . .
 ENV GO_PACKAGE github.com/openshift/service-ca-operator
 RUN make build --warn-undefined-variables \
     && gzip service-ca-operator-tests-ext
 
-FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
+FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
 COPY --from=builder /go/src/github.com/openshift/service-ca-operator/service-ca-operator /usr/bin/
 COPY --from=builder /go/src/github.com/openshift/service-ca-operator/service-ca-operator-tests-ext.gz /usr/bin/
 COPY manifests /manifests
