diff --git a/OWNERS b/OWNERS
index e6fb7a2b..89e4ff85 100644
--- a/OWNERS
+++ b/OWNERS
@@ -1,16 +1,20 @@
-approvers:
-- enxebre
-- csrwng
-- sjenning
-- muraee
+reviewers:
 - bryan-cox
-- jparrill
+- clebs
+- csrwng
 - devguyio
-reviewers:
 - enxebre
-- csrwng
-- sjenning
+- jparrill
 - muraee
+- Nirshal
+- sdminonne
+- sjenning
+approvers:
 - bryan-cox
-- jparrill
+- csrwng
 - devguyio
+- enxebre
+- jparrill
+- muraee
+- sjenning
+component: hypershift
