diff --git a/base/Dockerfile.rhel9 b/base/Dockerfile.rhel9
index 33130d56b..f5dee2077 100644
--- a/base/Dockerfile.rhel9
+++ b/base/Dockerfile.rhel9
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/4.22:base-rhel9-minimal
+FROM registry.ci.openshift.org/ocp/4.22:base-rhel9
 
 # A ubi9 image will expose python3 as /usr/bin/python. It does not contain
 # python2. Subsequent layers should install if it needed.
