diff --git a/Makefile b/Makefile
index f4cbf10051c..26efdfada8e 100644
--- a/Makefile
+++ b/Makefile
@@ -30,6 +30,10 @@ GITLINT_DIST_DIR := gitlint_dist
 GITLINT_BIN := gitlint
 GITLINT := $(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR)/$(GITLINT_BIN)-bin
 
+PYYAML_VER := 6.0.3
+PYYAML_DIST_DIR := pyyaml_dist
+PYYAML_STAMP := $(TOOLS_BIN_DIR)/$(PYYAML_DIST_DIR)/.installed
+
 PROMTOOL=$(abspath $(TOOLS_BIN_DIR)/promtool)
 
 # Setup envtest for running tests that require a Kubernetes API server
@@ -560,8 +564,8 @@ run-operator-locally-aws-dev:
 	@$(RUN_OPERATOR_LOCALLY_AWS)
 
 .PHONY: verify-docs-nav
-verify-docs-nav: ## Verify docs nav entries are sorted alphabetically.
-	python3 hack/verify-docs-nav-order.py
+verify-docs-nav: $(PYYAML_STAMP) ## Verify docs nav entries are sorted alphabetically.
+	PYTHONPATH=$(TOOLS_BIN_DIR)/$(PYYAML_DIST_DIR) python3 hack/verify-docs-nav-order.py
 
 .PHONY: verify-codespell
 verify-codespell: codespell ## Verify codespell.
@@ -600,6 +604,13 @@ karpenter-upstream-e2e:
 ## Tooling Binaries
 ## --------------------------------------
 
+##@ pyyaml
+$(PYYAML_STAMP): ## Install pyyaml for verify-docs-nav.
+	rm -rf $(TOOLS_BIN_DIR)/$(PYYAML_DIST_DIR) && \
+	mkdir -p $(TOOLS_BIN_DIR)/$(PYYAML_DIST_DIR) && \
+	python3 -m pip install --target=$(TOOLS_BIN_DIR)/$(PYYAML_DIST_DIR) pyyaml==$(PYYAML_VER) --upgrade && \
+	touch $@
+
 ##@ codespell
 codespell : $(CODESPELL) ## Build a local copy of codespell.
 $(CODESPELL): ## Build codespell from tools folder.
@@ -607,7 +618,7 @@ $(CODESPELL): ## Build codespell from tools folder.
 		mkdir -p $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR); \
 		mkdir -p $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR)/bin; \
 		mkdir -p $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR); \
-	 	pip install --target=$(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR) $(CODESPELL_BIN)==$(CODESPELL_VER) --upgrade; \
+	 	python3 -m pip install --target=$(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR) $(CODESPELL_BIN)==$(CODESPELL_VER) --upgrade; \
 		mv $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR)/bin/$(CODESPELL_BIN) $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR); \
 		rm -r $(TOOLS_BIN_DIR)/$(CODESPELL_DIST_DIR)/bin;
 
@@ -616,6 +627,6 @@ gitlint : $(GITLINT) ## Install local copy of gitlint
 $(GITLINT): $(TOOLS_DIR)/go.mod
 	mkdir -p $(TOOLS_BIN_DIR); \
 	mkdir -p $(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR); \
-	pip install --target=$(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR) gitlint==$(GITLINT_VER) --upgrade; \
+	python3 -m pip install --target=$(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR) gitlint==$(GITLINT_VER) --upgrade; \
 	cp $(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR)/bin/$(GITLINT_BIN) $(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR)/$(GITLINT_BIN)-bin; \
 	chmod +x $(TOOLS_BIN_DIR)/$(GITLINT_DIST_DIR)/$(GITLINT_BIN)-bin;
