diff --git a/test/e2e/v2/tests/backup_restore_test.go b/test/e2e/v2/tests/backup_restore_test.go
index 7adcbf6933d..aa130039bb6 100644
--- a/test/e2e/v2/tests/backup_restore_test.go
+++ b/test/e2e/v2/tests/backup_restore_test.go
@@ -72,6 +72,10 @@ var backupRestorePlatforms = map[hyperv1.PlatformType]backupRestorePlatformConfi
 		excludeWorkloads: []string{"router", "karpenter", "karpenter-operator", "cloud-network-config-controller"},
 		postRestoreHook:  nil,
 	},
+	hyperv1.KubevirtPlatform: {
+		excludeWorkloads: []string{"router", "karpenter", "karpenter-operator", "cloud-network-config-controller"},
+		postRestoreHook:  nil,
+	},
 }
 
 var _ = Describe("BackupRestore", Label("backup-restore"), Ordered, Serial, func() {
