diff --git a/.ci-operator.yaml b/.ci-operator.yaml
index e307e5af6..1d88a59fd 100644
--- a/.ci-operator.yaml
+++ b/.ci-operator.yaml
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.24-openshift-4.21
+  tag: rhel-9-release-golang-1.26-openshift-5.0
diff --git a/openshift-hack/images/Dockerfile.openshift b/openshift-hack/images/Dockerfile.openshift
index ddff40ee9..09d2d538e 100644
--- a/openshift-hack/images/Dockerfile.openshift
+++ b/openshift-hack/images/Dockerfile.openshift
@@ -1,10 +1,10 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.21 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.26-openshift-5.0 AS builder
 
 WORKDIR /build
 COPY . .
 RUN make build
 
-FROM registry.ci.openshift.org/ocp/4.21:base-rhel9
+FROM registry.ci.openshift.org/ocp/5.0:base-rhel9
 
 LABEL io.openshift.release.operator=true
 LABEL description="KubeVirt Cloud Controller Manager"
