diff --git a/hack/go-lint.sh b/hack/go-lint.sh
index c916284e8bf..9450887fdea 100755
--- a/hack/go-lint.sh
+++ b/hack/go-lint.sh
@@ -5,5 +5,5 @@ podman run --rm \
     --env IS_CONTAINER=TRUE \
     --volume "${PWD}:/go/src/github.com/openshift/installer:z" \
     --workdir /go/src/github.com/openshift/installer \
-    docker.io/golangci/golangci-lint:v2.3.1 \
-    golangci-lint run -c .golangci-lint-v2.yaml -v --new-from-rev=dcf8122 "${@}"
+    docker.io/golangci/golangci-lint:v2.10.1 \
+    golangci-lint run -c .golangci-lint-v2.yaml -v --new-from-rev=a6ba91c "${@}"
