diff --git a/assets/cluster-monitoring-operator/prometheus-rule.yaml b/assets/cluster-monitoring-operator/prometheus-rule.yaml
index c6a5c287be..ee59ca30f1 100644
--- a/assets/cluster-monitoring-operator/prometheus-rule.yaml
+++ b/assets/cluster-monitoring-operator/prometheus-rule.yaml
@@ -216,9 +216,25 @@ spec:
           )
         )
       record: cluster:kubelet_volume_stats_used_bytes:provisioner:sum
-    - expr: sum by (instance) (apiserver_storage_objects != -1)
+    - expr: sum by (instance) (apiserver_resource_objects != -1)
       record: instance:etcd_object_counts:sum
-    - expr: topk(500, max by(resource) (apiserver_storage_objects != -1))
+    - expr: |
+        topk(
+          500,
+          max by (resource) (
+              label_replace(
+                  label_join(apiserver_resource_objects{group!=""}, "resource_group", ".", "resource", "group")
+                or
+                  label_replace(apiserver_resource_objects{group=""}, "resource_group", "$1", "resource", "(.+)"),
+                "resource",
+                "$1",
+                "resource_group",
+                "(.+)"
+              )
+            !=
+              -1
+          )
+        )
       record: cluster:usage:resources:sum
     - expr: count(count (kube_pod_restart_policy{type!="Always",namespace!~"openshift-.+"}) by (namespace,pod))
       record: cluster:usage:pods:terminal:workload:sum
diff --git a/assets/control-plane/prometheus-rule.yaml b/assets/control-plane/prometheus-rule.yaml
index 5cc21b2657..6ce815d6f2 100644
--- a/assets/control-plane/prometheus-rule.yaml
+++ b/assets/control-plane/prometheus-rule.yaml
@@ -220,9 +220,16 @@ spec:
         description: HPA {{ $labels.namespace }}/{{ $labels.horizontalpodautoscaler  }} has been running at max replicas for longer than 15 minutes.
         summary: HPA is running at max replicas
       expr: |
-        kube_horizontalpodautoscaler_status_current_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
-          ==
-        kube_horizontalpodautoscaler_spec_max_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
+        (
+          kube_horizontalpodautoscaler_status_current_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
+            ==
+          kube_horizontalpodautoscaler_spec_max_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
+        )
+        and on(namespace, horizontalpodautoscaler) (
+          kube_horizontalpodautoscaler_spec_max_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
+            !=
+          kube_horizontalpodautoscaler_spec_min_replicas{namespace=~"(openshift-.*|kube-.*|default)",job="kube-state-metrics"}
+        )
       for: 15m
       labels:
         severity: warning
@@ -512,7 +519,9 @@ spec:
         runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/cluster-monitoring-operator/KubeletDown.md
         summary: Target disappeared from Prometheus target discovery.
       expr: |
-        absent(up{job="kubelet", metrics_path="/metrics"})
+        count by (cluster) (kube_node_info{job="kube-state-metrics"})
+        unless on (cluster)
+        count by (cluster) (up{job="kubelet", metrics_path="/metrics"} == 1)
       for: 15m
       labels:
         namespace: kube-system
@@ -783,10 +792,10 @@ spec:
         quantile: "0.99"
       record: cluster_quantile:scheduler_scheduling_algorithm_duration_seconds:histogram_quantile
     - expr: |
-        histogram_quantile(0.99, sum(rate(scheduler_binding_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
+        histogram_quantile(0.99, sum(rate(scheduler_pod_scheduling_sli_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
       labels:
         quantile: "0.99"
-      record: cluster_quantile:scheduler_binding_duration_seconds:histogram_quantile
+      record: cluster_quantile:scheduler_pod_scheduling_sli_duration_seconds:histogram_quantile
     - expr: |
         histogram_quantile(0.9, sum(rate(scheduler_scheduling_attempt_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
       labels:
@@ -798,10 +807,10 @@ spec:
         quantile: "0.9"
       record: cluster_quantile:scheduler_scheduling_algorithm_duration_seconds:histogram_quantile
     - expr: |
-        histogram_quantile(0.9, sum(rate(scheduler_binding_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
+        histogram_quantile(0.9, sum(rate(scheduler_pod_scheduling_sli_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
       labels:
         quantile: "0.9"
-      record: cluster_quantile:scheduler_binding_duration_seconds:histogram_quantile
+      record: cluster_quantile:scheduler_pod_scheduling_sli_duration_seconds:histogram_quantile
     - expr: |
         histogram_quantile(0.5, sum(rate(scheduler_scheduling_attempt_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
       labels:
@@ -813,10 +822,10 @@ spec:
         quantile: "0.5"
       record: cluster_quantile:scheduler_scheduling_algorithm_duration_seconds:histogram_quantile
     - expr: |
-        histogram_quantile(0.5, sum(rate(scheduler_binding_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
+        histogram_quantile(0.5, sum(rate(scheduler_pod_scheduling_sli_duration_seconds_bucket{job="scheduler"}[5m])) without(instance, pod))
       labels:
         quantile: "0.5"
-      record: cluster_quantile:scheduler_binding_duration_seconds:histogram_quantile
+      record: cluster_quantile:scheduler_pod_scheduling_sli_duration_seconds:histogram_quantile
   - name: node.rules
     rules:
     - expr: |
diff --git a/assets/control-plane/service-monitor-kubelet.yaml b/assets/control-plane/service-monitor-kubelet.yaml
index 408613c86e..7ba1d53048 100644
--- a/assets/control-plane/service-monitor-kubelet.yaml
+++ b/assets/control-plane/service-monitor-kubelet.yaml
@@ -27,7 +27,7 @@ spec:
       sourceLabels:
       - __name__
     - action: drop
-      regex: apiserver_(request_count|request_latencies|request_latencies_summary|dropped_requests|storage_data_key_generation_latencies_microseconds|storage_transformation_failures_total|storage_transformation_latencies_microseconds|proxy_tunnel_sync_latency_secs|longrunning_gauge|registered_watchers|storage_db_total_size_in_bytes|flowcontrol_request_concurrency_limit|flowcontrol_request_concurrency_in_use)
+      regex: apiserver_(request_count|request_latencies|request_latencies_summary|dropped_requests|storage_data_key_generation_latencies_microseconds|storage_transformation_failures_total|storage_transformation_latencies_microseconds|proxy_tunnel_sync_latency_secs|longrunning_gauge|registered_watchers|storage_db_total_size_in_bytes|flowcontrol_request_concurrency_limit|flowcontrol_request_concurrency_in_use|storage_objects)
       sourceLabels:
       - __name__
     - action: drop
diff --git a/assets/node-exporter/prometheus-rule.yaml b/assets/node-exporter/prometheus-rule.yaml
index dfb675d6c3..5d128483bc 100644
--- a/assets/node-exporter/prometheus-rule.yaml
+++ b/assets/node-exporter/prometheus-rule.yaml
@@ -354,6 +354,26 @@ spec:
           rate(node_network_transmit_drop_total{job="node-exporter", device!="lo"}[1m])
         )
       record: instance:node_network_transmit_drop_excluding_lo:rate1m
+    - expr: |
+        sum without (device) (
+          rate(node_network_receive_bytes_total{job="node-exporter", device!~"lo|veth.+"}[1m])
+        )
+      record: instance:node_network_receive_bytes_physical:rate1m
+    - expr: |
+        sum without (device) (
+          rate(node_network_transmit_bytes_total{job="node-exporter", device!~"lo|veth.+"}[1m])
+        )
+      record: instance:node_network_transmit_bytes_physical:rate1m
+    - expr: |
+        sum without (device) (
+          rate(node_network_receive_drop_total{job="node-exporter", device!~"lo|veth.+"}[1m])
+        )
+      record: instance:node_network_receive_drop_physical:rate1m
+    - expr: |
+        sum without (device) (
+          rate(node_network_transmit_drop_total{job="node-exporter", device!~"lo|veth.+"}[1m])
+        )
+      record: instance:node_network_transmit_drop_physical:rate1m
   - name: telemetry
     rules:
     - expr: sum by(vendor,model) (node_accelerator_card_info)
diff --git a/jsonnet/jsonnetfile.lock.json b/jsonnet/jsonnetfile.lock.json
index 7a4de5dd6e..79b7468013 100644
--- a/jsonnet/jsonnetfile.lock.json
+++ b/jsonnet/jsonnetfile.lock.json
@@ -18,7 +18,7 @@
           "subdir": "contrib/mixin"
         }
       },
-      "version": "6775d033bc560a4604831837e56cc914e3fe172d",
+      "version": "be3acf4ed017efb7008beb08585d890e5364ce93",
       "sum": "XmXkOCriQIZmXwlIIFhqlJMa0e6qGWdxZD+ZDYaN0Po="
     },
     {
@@ -28,8 +28,8 @@
           "subdir": "grafana-mixin"
         }
       },
-      "version": "3cdec6626521b1126db47c3b816ffdc485f1291f",
-      "sum": "S8mRTRH4w62kMCa2je3iCtvscYrwQmkyJ7Y/aM14KbE="
+      "version": "d45eac04111b8d1e875a5335d12ab3d4fde8814f",
+      "sum": "sRjOJwJhhkCDmMQls7clo/csVix/b47W+Wc3YCzn64Y="
     },
     {
       "source": {
@@ -48,7 +48,7 @@
           "subdir": "gen/grafonnet-latest"
         }
       },
-      "version": "8a27651b56fbdba45a389ccb11440200091c73a1",
+      "version": "7380c9c64fb973f34c3ec46265621a2b0dee0058",
       "sum": "V9vAj21qJOc2DlMPDgB1eEjSQU4A+sAA4AXuJ6bd4xc="
     },
     {
@@ -58,7 +58,7 @@
           "subdir": "gen/grafonnet-v10.0.0"
         }
       },
-      "version": "8a27651b56fbdba45a389ccb11440200091c73a1",
+      "version": "7380c9c64fb973f34c3ec46265621a2b0dee0058",
       "sum": "xdcrJPJlpkq4+5LpGwN4tPAuheNNLXZjE6tDcyvFjr0="
     },
     {
@@ -68,7 +68,7 @@
           "subdir": "gen/grafonnet-v11.4.0"
         }
       },
-      "version": "8a27651b56fbdba45a389ccb11440200091c73a1",
+      "version": "7380c9c64fb973f34c3ec46265621a2b0dee0058",
       "sum": "aVAX09paQYNOoCSKVpuk1exVIyBoMt/C50QJI+Q/3nA="
     },
     {
@@ -78,8 +78,8 @@
           "subdir": "grafana-builder"
         }
       },
-      "version": "d1eede9462c03d44dcd4a707aa8e7dcaf155a552",
-      "sum": "d8iL8gbyQxspZJ8mDGsqHRR/JF3cu+YOaV50vao93Nw="
+      "version": "2e9c7f4a276f6f6a9bc3f994f2b8804555f31bb9",
+      "sum": "TWpxm5vPO23V3hoF3pdwGs1NvR+PQce9baN716mJq+4="
     },
     {
       "source": {
@@ -88,8 +88,8 @@
           "subdir": "mixin-utils"
         }
       },
-      "version": "d1eede9462c03d44dcd4a707aa8e7dcaf155a552",
-      "sum": "VAik6Sh5MD5H1Km1gSIXG4rwQ4m4zyw7odP5TKu3bGo="
+      "version": "2e9c7f4a276f6f6a9bc3f994f2b8804555f31bb9",
+      "sum": "zVkvwae4zHP6xeIoDvde/wldI6ClprFjnk50mMD9/z4="
     },
     {
       "source": {
@@ -129,8 +129,8 @@
           "subdir": ""
         }
       },
-      "version": "cec7af386c5d615a388f8101430a3b782d220d6c",
-      "sum": "kTICCjpL3vAC4yIKufo/Gd9YaTTa9MiWS0EDGC/loUM="
+      "version": "644bc5c6ab9dd5583bb04f76ef3d320fc3f0f2b3",
+      "sum": "XJNzVA/gK8R4ORRKQXJomW56T3QloPmbgSqi/R4cxJ0="
     },
     {
       "source": {
@@ -139,8 +139,8 @@
           "subdir": "jsonnet/kube-state-metrics"
         }
       },
-      "version": "9ea375226bc72c0283bd0d35a7d4b3e399749b73",
-      "sum": "3bioG7CfTfY9zeu5xU4yon6Zt3kYvNkyl492nOhQxnM="
+      "version": "0742699d15c34feb4169278dfae1dd2670d95d33",
+      "sum": "QimcNpwZcRriOp64HdmGCJuV6mzDs9FPWJ7XFnko3Yo="
     },
     {
       "source": {
@@ -149,7 +149,7 @@
           "subdir": "jsonnet/kube-state-metrics-mixin"
         }
       },
-      "version": "9ea375226bc72c0283bd0d35a7d4b3e399749b73",
+      "version": "0742699d15c34feb4169278dfae1dd2670d95d33",
       "sum": "qclI7LwucTjBef3PkGBkKxF0mfZPbHnn4rlNWKGtR4c="
     },
     {
@@ -159,7 +159,7 @@
           "subdir": "jsonnet"
         }
       },
-      "version": "ab605a4bf1c6cd5a6c58f245fb67ce648caebfa8",
+      "version": "e71e09969f11a47c87b87c43b762ad1d01f6b04f",
       "sum": "av+OVY/HFS87sJGoH3TJ7f0QkGZls/KmaAnMYw3Ou8Q=",
       "name": "openshift-state-metrics"
     },
@@ -170,8 +170,8 @@
           "subdir": "jsonnet/telemeter"
         }
       },
-      "version": "80acf50b0b105f9a83b2caec8a1e9aba05ab0f4d",
-      "sum": "sgPbiGdB2J/VDC14OeDwKK8Okru2gYRKygv+urKYuhA=",
+      "version": "d6c7dd499f7106df78aa8bfd9dee332121ab2f09",
+      "sum": "F3o5CyKp4I6z+c1DV2EkRT8QipF+EhRRbUp6wSN9ev8=",
       "name": "telemeter-client"
     },
     {
@@ -181,8 +181,8 @@
           "subdir": "jsonnet/kube-prometheus"
         }
       },
-      "version": "2e016283f2505a99ed03687b36d41d3e9dd5c49f",
-      "sum": "ky9yNtgZxHDmXiygPo6pJl1GzBOiRfuGKmUEeiv+0rA="
+      "version": "b62e3aed2a882b2d13363ff63103987f27228f71",
+      "sum": "tYWiqTO1LFVfBRfM86YtJwhSIan8Fq5RwXO/SylGnCA="
     },
     {
       "source": {
@@ -212,8 +212,8 @@
           "subdir": "doc/alertmanager-mixin"
         }
       },
-      "version": "f7b4feffd3c8d4570e7daed234c4ba56b9ef8b05",
-      "sum": "j5prvRrJdoCv7n45l5Uy2ghl1IDb9BBUqjwCDs4ZJoQ=",
+      "version": "7aa1b9f902569e936244e4ae77be1aab47e608b3",
+      "sum": "diZM7b/jQKqaIjuVSiseRaxgpgdQyVCtCgh0XhyPnXQ=",
       "name": "alertmanager"
     },
     {
@@ -223,8 +223,8 @@
           "subdir": "docs/node-mixin"
         }
       },
-      "version": "b306b639e1c81dfd6e142d5415a85b37c9f1c195",
-      "sum": "t5lGWjS4xUmXbOaMg8BddbJrIJeglTIi8HkT3UBWtP4="
+      "version": "0f5c15834bcfe027be499abe34673e869d4d5386",
+      "sum": "PGx1okd1Ny4U+Ma3AV0haZWvXnX8eJBN4+BsQInWUo4="
     },
     {
       "source": {
@@ -233,7 +233,7 @@
           "subdir": "documentation/prometheus-mixin"
         }
       },
-      "version": "d0d2699dc5f454ebca53e56247281447f8dd8c5a",
+      "version": "7df2d13f009ed2ac2c9a742fbf3f5c9edf3845af",
       "sum": "UuADJwhNploRwfJYGb0sr2mi+MvswKpumoOMXnGs0Dc=",
       "name": "prometheus"
     },
@@ -244,8 +244,8 @@
           "subdir": "jsonnet/controller-gen"
         }
       },
-      "version": "6f16e091a98c44792808d238f1b7db1a97ead23a",
-      "sum": "xUGxDjMr7V/SpY21Xsr0X3ve/a04du3pRlbrA3I/Uk8=",
+      "version": "91b222901fe48185861beefafc4c671c1f63d250",
+      "sum": "uIAr+mflJJhdKR4DvXZV9+E+Sbvt38CPE2nS4RrAsWU=",
       "name": "pyrra"
     },
     {
@@ -255,8 +255,8 @@
           "subdir": "jsonnet/kube-thanos"
         }
       },
-      "version": "6fedb045db2aeb0a4a880f77bfdfb5d4580f51f9",
-      "sum": "cCbGOfWt+5lh8FTjLBB1w1815K1UUZ9g+C8CQruNwFY="
+      "version": "55d4a7c28b2b6c8310cee030c6b319868ec9bd62",
+      "sum": "1dYP6hnUX/+rqIWm9FSVQ1qHGCk3np/8ss0gFdNnjHs="
     },
     {
       "source": {
@@ -265,7 +265,7 @@
           "subdir": "mixin"
         }
       },
-      "version": "b3b08ef8d13538b603f0a7af917ccb6960a71b7e",
+      "version": "aec3c15b6803a4c24286ad3d3a9dca6f357657aa",
       "sum": "ieCD4eMgGbOlrI8GmckGPHBGQDcLasE1rULYq56W/bs="
     }
   ],
diff --git a/jsonnet/rules.libsonnet b/jsonnet/rules.libsonnet
index 2546d4f8cd..7704b3c54d 100644
--- a/jsonnet/rules.libsonnet
+++ b/jsonnet/rules.libsonnet
@@ -319,14 +319,34 @@ function(params) {
           // This recording rule doesn't expose objects count of -1 as it
           // indicates issues between the kube-apiserver and etcd and pollutes
           // the data in telemetry.
-          expr: 'sum by (instance) (apiserver_storage_objects != -1)',
+          expr: 'sum by (instance) (apiserver_resource_objects != -1)',
           record: 'instance:etcd_object_counts:sum',
         },
         {
           // This recording rule doesn't expose objects count of -1 as it
           // indicates issues between the kube-apiserver and etcd and pollutes
           // the data in telemetry.
-          expr: 'topk(500, max by(resource) (apiserver_storage_objects != -1))',
+          // The expression concatenates the resource and group labels together
+          // because this is how the original recorded metric was constructed.
+          // Core resources have an empty group label hence the 'or' clause.
+          expr: |||
+            topk(
+              500,
+              max by (resource) (
+                  label_replace(
+                      label_join(apiserver_resource_objects{group!=""}, "resource_group", ".", "resource", "group")
+                    or
+                      label_replace(apiserver_resource_objects{group=""}, "resource_group", "$1", "resource", "(.+)"),
+                    "resource",
+                    "$1",
+                    "resource_group",
+                    "(.+)"
+                  )
+                !=
+                  -1
+              )
+            )
+          |||,
           record: 'cluster:usage:resources:sum',
         },
         {
diff --git a/test/e2e/telemeter_test.go b/test/e2e/telemeter_test.go
index 72c3de388f..51f4af8766 100644
--- a/test/e2e/telemeter_test.go
+++ b/test/e2e/telemeter_test.go
@@ -116,3 +116,31 @@ func TestTelemeterClient(t *testing.T) {
 		)
 	}
 }
+
+// TestTelemeterClusterMetrics verifies that key Telemetry metrics are present.
+func TestTelemeterClusterMetrics(t *testing.T) {
+	for _, q := range []string{
+		`count(cluster:usage:resources:sum) > 0`,
+		`count(cluster:usage:resources:sum{resource="pods"}) > 0`,
+		`count(cluster:usage:resources:sum{resource="deployments.apps"}) > 0`,
+		`count(cluster:usage:resources:sum{resource="clusterroles.rbac.authorization.k8s.io"}) > 0`,
+		`count(count:up1) > 0`,
+		`count(node_role_os_version_machine:cpu_capacity_cores:sum) > 0`,
+		`count(node_role_os_version_machine:cpu_capacity_sockets:sum) > 0`,
+		`count(cluster:cpu_usage_cores:sum) > 0`,
+		`count(cluster:memory_usage_bytes:sum) > 0`,
+	} {
+		t.Run(q, func(t *testing.T) {
+			t.Parallel()
+			f.PrometheusK8sClient.WaitForQueryReturn(
+				t,
+				time.Minute,
+				q,
+				func(float64) error {
+					// Queries are written to return something if the expected condition is met.
+					return nil
+				},
+			)
+		})
+	}
+}
