diff --git a/test/extended/cli/timeout.go b/test/extended/cli/timeout.go
index a95e54138717..f1937b5526b1 100644
--- a/test/extended/cli/timeout.go
+++ b/test/extended/cli/timeout.go
@@ -25,11 +25,11 @@ var _ = g.Describe("[sig-cli] oc --request-timeout", func() {
 		// from server or from context it's ok
 		o.Expect(out).Should(o.SatisfyAny(o.ContainSubstring("request canceled"), o.ContainSubstring("context deadline exceeded")))
 
-		out, err = oc.Run("get", "dc/testdc").Args("--request-timeout=2s").Output()
+		out, err = oc.Run("get", "dc/testdc").Args("--request-timeout=5s").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(out).To(o.ContainSubstring("testdc"))
 
-		out, err = oc.Run("get", "dc/testdc").Args("--request-timeout=2").Output()
+		out, err = oc.Run("get", "dc/testdc").Args("--request-timeout=5").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(out).To(o.ContainSubstring("testdc"))
 
@@ -53,11 +53,11 @@ var _ = g.Describe("[sig-cli] oc --request-timeout", func() {
 		// from server or from context it's ok
 		o.Expect(out).Should(o.SatisfyAny(o.ContainSubstring("request canceled"), o.ContainSubstring("context deadline exceeded")))
 
-		out, err = oc.Run("get", "deployment/testdc").Args("--request-timeout=2s").Output()
+		out, err = oc.Run("get", "deployment/testdc").Args("--request-timeout=5s").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(out).To(o.ContainSubstring("testdc"))
 
-		out, err = oc.Run("get", "deployment/testdc").Args("--request-timeout=2").Output()
+		out, err = oc.Run("get", "deployment/testdc").Args("--request-timeout=5").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(out).To(o.ContainSubstring("testdc"))
 
